---
title: "2023_01_05_formalin_fixed_measles_reads_lung_only"
author: "KM"
date: '2023-01-05'
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r Load-libraries}
library("plotly"); packageVersion("plotly")
library("ggpubr"); packageVersion("ggpubr")
library("cowplot"); packageVersion("cowplot")
library("glue"); packageVersion("glue")
library("ggrepel"); packageVersion("ggrepel")
library("ggsci"); packageVersion("ggsci")
library("lubridate"); packageVersion("lubridate")
library("tidylog"); packageVersion("tidylog")
library("data.table"); packageVersion("data.table")
library("phyloseq"); packageVersion("phyloseq")
library("speedyseq"); packageVersion("speedyseq")
library("plyr"); packageVersion("plyr")
library("tidyverse"); packageVersion("tidyverse")
library("rstatix"); packageVersion("rstatix")
library("ComplexHeatmap"); packageVersion("ComplexHeatmap")
library("scales"); packageVersion("scales")

# Set a random seed so that exact results can be reproduced
set.seed(1000)
```


# Import Read Data
```{r import-data-reads}
# bigtable all; extract viruses
df.virus.reads <- as_tibble(fread(file = "~/Handley Lab Dropbox/16S/Celiac/hecatomb_output/bigtable.tsv", header = TRUE))
dim(df.virus.reads)
# 115738102       32

sapply(strsplit(df.virus.reads$sampleID,"_"))
View(head(df.virus.reads))

length(unique(df.virus.reads[["sampleID"]]))
# 336

length(unique(df.virus.reads[["seqID"]]))
# 15738102

#write.table(df.virus.reads, file = "./lung_only/reads/2023_01_05_formalin_fixed_measles_lung_only_viral_bigtable.tsv", sep = "\t", row.names = FALSE, quote=FALSE)

# Select only "bacterial" reads
# df.nonviral.reads <- as_tibble(fread(file = "./lung_only/data/bigtable.nonviral.tsv", header = TRUE))
# dim(df.nonviral.reads)
# # 32,417    32
# 
# df.bac.reads <- df.nonviral.reads %>%
#   filter(kingdom == "Bacteria")
# dim(df.bac.reads)
# # 31,345 x 32
# 
# length(unique(df.bac.reads[["seqID"]]))
# # 31,119
# 
# write.table(df.bac.reads, file = "./lung_only/reads/2023_01_05_formalin_fixed_measles_lung_only_bacterial_bigtable.tsv", sep = "\t", row.names = FALSE, quote=FALSE)
# 
# rm(df.bac.reads); gc()
# 
# # Dark matter
# df.dark.reads <- df.nonviral.reads %>%
#   filter(kingdom == "unclassified  superkingdom" | kingdom == "unclassified unclassified entries superkingdom")
# dim(df.dark.reads)
# # 101  32
# 
# length(unique(df.dark.reads[["seqID"]]))
# # 101
# 
# write.table(df.dark.reads, file = "./lung_only/reads/2023_01_05_formalin_fixed_measles_lung_only_dark_matter_bigtable.tsv", sep = "\t", row.names = FALSE, quote=FALSE)

# Create column for database_ID, based on sampleID
samp.d <- fread("/Users/handley_lab/Handley Lab Dropbox/virome/viral_archaeology/formalin_fixed_measles/Formalin_fixed_measles_viral_bigtable_sample_ID.txt")

# Import metadata
MAP <- import_qiime_sample_data("~/Handley Lab Dropbox/16S/Celiac/celiacs_metadata.txt")
celiac_meta <- fread("~/Handley Lab Dropbox/16S/Celiac/celiacs_metadata_final.csv")

celiac_meta$`Study ID`

df.virus.reads$sampleID
View(celiac_meta)
# Convert MAP to data.frame and remove row.names
map.df <- as(sample_data(MAP), "data.frame") %>% 
  `rownames<-`( NULL )

# Merge metadata with viral bigtable
df.virus.tmp <- merge(df.virus.reads, samp.d, by = "sampleID")
dim(df.virus.tmp)
# [1] 11,189  33

df.virus.fin.reads <- left_join(df.virus.tmp, map.df, by = c("database_ID", "sampleID"))
dim(df.virus.fin.reads)
# [1] 11,189  55

write.table(df.virus.fin.reads, file = "./lung_only/reads/2023_01_05_formalin_fixed_measles_lung_only_viral_meta_bigtable.tsv", sep = "\t", row.names = FALSE, quote=FALSE)

rm(df.virus.tmp, samp.d); gc()

# Start from viral table with metadata: 
df.virus.fin.reads <- fread(file = "./lung_only/reads/2023_01_05_formalin_fixed_measles_lung_only_viral_meta_bigtable.tsv", header = TRUE, stringsAsFactors = TRUE, nThread = 4)

```



```{r add-function-to-word-wrap}
# Default 15 character target width.
swr = function(string, nwrap=15) {
  paste(strwrap(string, width=nwrap), collapse="\n")
}

swr = Vectorize(swr)

# Default 20 character target width.
swr20 = function(string, nwrap=20) {
  paste(strwrap(string, width=nwrap), collapse="\n")
}

swr20 = Vectorize(swr20)

# Default 30 character target width.
swr30 = function(string, nwrap=30) {
  paste(strwrap(string, width=nwrap), collapse="\n")
}

swr30 = Vectorize(swr30)
```


# aesthetics
```{r aesthetics}
# Color by alignment type
alnType.cols  <- c("aa" = "#ff7f50", "nt" = "#00BFC4") 
 # aa = coral = #ff7f50;  nt = #00BFC4 = Turquoise Surf

```

# Glom at family, genus and species levels
```{r glom-tax-levels}
# reads
df.virus.fam.reads <- df.virus.fin.reads %>%
  unite("lineage", c("kingdom", "phylum", "class", "order", "family"), sep = ";", remove = FALSE) %>%
  group_by(sampleID, lineage, family) %>%
  mutate(sum = sum(count)) %>%
  separate("lineage", c("kingdom", "phylum", "class", "order", "family"), sep = ";") %>%
  select(1,24:28,31:56) %>%
  distinct()

df.virus.genus.reads <- df.virus.fin.reads %>%
  unite("lineage", c("kingdom", "phylum", "class", "order", "family", "genus"), sep = ";", remove = FALSE) %>%
  group_by(sampleID, lineage, genus) %>%
   mutate(sum = sum(count))  %>%
  separate("lineage", c("kingdom", "phylum", "class", "order", "family", "genus"), sep = ";") %>%
  select(1,24:29,31:56) %>%
  distinct()

df.virus.species.reads <- df.virus.fin.reads %>%
  unite("lineage", c("kingdom", "phylum", "class", "order", "family", "genus", "species"), sep = ";", remove = FALSE) %>%
  group_by(sampleID, lineage, species) %>%
   mutate(sum = sum(count))  %>%
  separate("lineage", c("kingdom", "phylum", "class", "order", "family", "genus", "species"), sep = ";") %>%
  select(1,24:56) %>%
  distinct()

```

# Initial Evaluation Reads
```{r evaluation-reads}
df.virus.fin.reads.fam.sample <- df.virus.fin.reads %>%
  group_by(sampleID, family, alnType) %>%
  summarise(count = n()) %>%
  filter(count > 10) %>%
  arrange(-count)

p.df.virus.fin.reads.fam.sample <- ggplot(df.virus.fin.reads.fam.sample, aes (x = sampleID, y = count, color = alnType)) +
  geom_point() +
  scale_y_log10() +
  facet_wrap(~swr20(family)) +
  theme(strip.text.x = element_text(size = 8)) +
  theme(axis.text.x = element_blank())

p.df.virus.fin.reads.fam.sample

# Plot
ggsave(p.df.virus.fin.reads.fam.sample, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_virus_reads_fam_sample.pdf', width = 10, height = 8, dpi = 600)

df.virus.fin.reads.fam.raw <- df.virus.fin.reads %>%
  group_by(family, alnType) %>%
  summarise(count = n()) %>%
  arrange(family)

# Get the number of reads by alnType, filling in 0 for missing values
df.virus.fin.reads.fam <- df.virus.fin.reads %>%
  group_by(family, alnType) %>%
  summarise(count = n()) %>%
  arrange(family) %>%
  ungroup() %>%
  complete(family, alnType, fill = list(count = 0)) %>%
  group_by(family, alnType) 

write.table(df.virus.fin.reads.fam, file = "./lung_only/report/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_family_counts.txt", sep = "\t", row.names = FALSE, quote=FALSE)

# Get the aa to nt alignment ratios
df.virus.fin.reads.fam.ratio <- df.virus.fin.reads.fam %>%
  pivot_wider(names_from = "alnType", values_from ="count") %>%
  mutate(aa_nt_ratio = sprintf("%.5f", aa / nt)) 

df.virus.fin.reads.fam.ratio$aa_nt_ratio <- as.numeric(df.virus.fin.reads.fam.ratio$aa_nt_ratio)

write.table(df.virus.fin.reads.fam.ratio, file = "./lung_only/report/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_family_ratio.txt", sep = "\t", row.names = FALSE, quote=FALSE)

# Look at the pident and alnlen overall
df.virus.fin.reads.family.a <- df.virus.fin.reads %>%
  group_by(family)  %>%
  mutate(Count = n()) 

# Plot
p.scatter.all.reads.fam <- ggplot(df.virus.fin.reads.family.a, aes(x = alnlen, y = pident, color = alnType)) +
  geom_jitter(position = position_jitter(width = 0.05, height = 0.05), size = 0.5, alpha = 0.5) +
  facet_wrap(facets = ~fct_reorder(swr20(family), Count, .fun = "median", .desc = TRUE)) +
  geom_hline(aes(yintercept = 70), lty = 2, color = "gray") + # strange ggplot2 behaviour. Have to add this as an aes() if using fct_reorder
  geom_vline(aes(xintercept = 150), lty = 2, color = "gray") +
  labs(x = "Alignment Length",
       y = "% ID",
       color = "Query Type",
       caption = "%ID and alignment length for all detected viral families.\n Dashed lines indicate 70% ID (y-axis) and 150 bases amino acids (x-axis)") +
  theme(axis.text = element_text(size = 8)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(strip.text = element_text(size = 7))

p.scatter.all.reads.fam

ggsave(p.scatter.all.reads.fam, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_family_scatterplot.pdf', width = 9, height = 9, dpi = 600)

# NT filter testing - determine an evalue cutoff using Paramyxoviridae as a model - YAH
df.virus.fin.reads.para <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Paramyxoviridae") %>%
  droplevels() %>%
  mutate(Count = n()) 
dim(df.virus.fin.reads.para)
# [1] 2136   56

# Plot
p.scatter.reads.para <- ggplot(df.virus.fin.reads.para, aes(x = alnlen, y = pident, color = alnType)) +
  geom_jitter(position = position_jitter(width = 0.05, height = 0.05), size = 1.0, alpha = 1.0) +
#  geom_hline(aes(yintercept = 70), lty = 2, color = "gray") + # strange ggplot2 behaviour. Have to add this as an aes() if using fct_reorder
  geom_vline(aes(xintercept = 150), lty = 2, color = "gray") +
  labs(x = "Alignment Length",
       y = "% ID",
       color = "Query Type",
       title = "Paramyxoviridae: All") +
  theme(axis.text = element_text(size = 8)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(strip.text = element_text(size = 7))

p.scatter.reads.para

ggsave(p.scatter.reads.para, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_scatterplot_para_all_evalue.pdf', width = 5, height = 3, dpi = 600)

# <= 1e-10
df.virus.fin.reads.para.10 <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Paramyxoviridae") %>%
  filter(alnType == "aa" | (alnType == "nt" & evalue <=1e-10)) %>%
  droplevels() %>%
  mutate(Count = n()) 
dim(df.virus.fin.reads.para.10)
# [1] 2136   56

tmp <- df.virus.fin.reads.para.10 %>% filter(alnType == "nt")
max(tmp$evalue)
# [1] 9.695e-21

# <= 1e-30
df.virus.fin.reads.para.30 <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Paramyxoviridae") %>%
  filter(alnType == "aa" | (alnType == "nt" & evalue <=1e-30)) %>%
  droplevels() %>%
  mutate(Count = n()) 
dim(df.virus.fin.reads.para.30)
# [1] 1762   56 -> loss of 374 reads!

# Plot
p.scatter.reads.para.30 <- ggplot(df.virus.fin.reads.para.30, aes(x = alnlen, y = pident, color = alnType)) +
  geom_jitter(position = position_jitter(width = 0.05, height = 0.05), size = 1.0, alpha = 1.0) +
#  geom_hline(aes(yintercept = 70), lty = 2, color = "gray") + # strange ggplot2 behaviour. Have to add this as an aes() if using fct_reorder
  geom_vline(aes(xintercept = 150), lty = 2, color = "gray") +
  labs(x = "Alignment Length",
       y = "% ID",
       color = "Query Type",
       title = "Paramyxoviridae: <=1e-30") +
  theme(axis.text = element_text(size = 8)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(strip.text = element_text(size = 7))

p.scatter.reads.para.30

ggsave(p.scatter.reads.para.30, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_scatterplot_para_all_evalue30.pdf', width = 5, height = 3, dpi = 600)

# Conclusion set nt filter of <=1e-30 - HOLD because ancient samples
df.virus.fin.reads.30 <- df.virus.fin.reads %>%
  filter(alnType == "aa" | (alnType == "nt" & evalue <=1e-30)) %>%
  droplevels() 

dim(df.virus.fin.reads)
# [1] 11,189    55
dim(df.virus.fin.reads.30)
# [1] 8,568    55

# Replot families with nt cutoff
df.virus.fin.reads.family.30 <- df.virus.fin.reads.30 %>%
  group_by(family)  %>%
  mutate(Count = n()) 

# Plot
p.scatter.all.fam.30 <- ggplot(df.virus.fin.reads.family.30, aes(x = alnlen, y = pident, color = alnType)) +
  geom_jitter(position = position_jitter(width = 0.05, height = 0.05), size = 1.0, alpha = 1.0) +
  facet_wrap(facets = ~fct_reorder(family, Count, .fun = "median", .desc = TRUE)) +
  geom_hline(aes(yintercept = 70), lty = 2, color = "gray") + # strange ggplot2 behaviour. Have to add this as an aes() if using fct_reorder
  geom_vline(aes(xintercept = 150), lty = 2, color = "gray") +
  labs(x = "Alignment Length",
       y = "% ID",
       color = "Query Type",
       caption = "%ID and alignment length for all detected viral families.\n Dashed lines indicate 70% ID (y-axis) and 150 bases amino acids (x-axis)") +
  theme(axis.text = element_text(size = 8)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(strip.text = element_text(size = 7))

p.scatter.all.fam.30

ggsave(p.scatter.all.fam.30, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_family_scatterplot_nt_evalue30.pdf', width = 9, height = 7, dpi = 600)

# Look at subset of foi (non-phage)
df.virus.fin.reads.family.comp <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter( family == "Adenoviridae" |  family == "Birnaviridae" | family == "Coronaviridae" | family == "Herpesviridae"  | family == "Orthomyxoviridae"  | family == "Paramyxoviridae" | family == "Picobirnaviridae" | family == "Pneumoviridae" | family == "Reoviridae"  | family == "Retroviridae" | family == "unclassified Picornavirales family"  | family == "unclassified Viruses family"  | family == "Virgaviridae" ) %>%
  droplevels() %>%
  mutate(Count = n()) 
  
# Plot
p.scatter.all.fam.comp <- ggplot(df.virus.fin.reads.family.comp, aes(x = alnlen, y = pident, color = alnType)) +
  geom_jitter(position = position_jitter(width = 0.05, height = 0.05), size = 0.5, alpha = 0.5) +
  facet_wrap(~family) +
  geom_hline(aes(yintercept = 70), lty = 2, color = "gray") + # strange ggplot2 behaviour. Have to add this as an aes() if using fct_reorder
  geom_vline(aes(xintercept = 150), lty = 2, color = "gray") +
  labs(x = "Alignment Length",
       y = "% ID",
       color = "Query Type",
       caption = "%ID and alignment length for all detected viral families.\n Dashed lines indicate 70% ID (y-axis) and 150 bases amino acids (x-axis)",
       title = "Reads") +
  theme(axis.text = element_text(size = 8)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(strip.text = element_text(size = 7))

p.scatter.all.fam.comp

ggsave(p.scatter.all.fam.comp, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_family_scatterplot_comp.pdf', width = 7, height = 5, dpi = 600)

# Look at unclassified Viruses Eukaryotic species - require at least 10 reads OR picorna
desired.euk.species <- c("Beihai picorna-like virus 14", "Beihai picorna-like virus 15", "Beihai picorna-like virus 17", "Beihai picorna-like virus 20", "Beihai picorna-like virus 21", "Beihai picorna-like virus 37", "Beihai picorna-like virus 4", "Beihai picorna-like virus 52", "Beihai picorna-like virus 53", "Sanxia picorna-like virus 3 ", "Sanxia picorna-like virus 7", "Shahe picorna-like virus 7", "Sylvanvirus sp.", "Wenzhou picorna-like virus 5", "Wenzhou picorna-like virus 6", "Wenzhou picorna-like virus 7", "Wenzhou tombus-like virus 2", "Riiser virus")

df.virus.fin.reads.species.comp <- df.virus.fin.reads %>%
  group_by(family,genus, species)  %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species %in% desired.euk.species) %>%
  droplevels() %>%
  mutate(Count = n()) 

# Plot
p.scatter.allv.euk.species.comp <- ggplot(df.virus.fin.reads.species.comp, aes(x = alnlen, y = pident, color = alnType)) +
  geom_jitter(position = position_jitter(width = 0.05, height = 0.05), size = 0.5, alpha = 0.5) +
  facet_wrap(~swr20(species)) +
  geom_hline(aes(yintercept = 70), lty = 2, color = "gray") + # strange ggplot2 behaviour. Have to add this as an aes() if using fct_reorder
  geom_vline(aes(xintercept = 150), lty = 2, color = "gray") +
  labs(x = "Alignment Length",
       y = "% ID",
       color = "Query Type",
       caption = "%ID and alignment length for unclassified Viruses family; eukaryotic viruses.\n Dashed lines indicate 70% ID (y-axis) and 150 bases amino acids (x-axis)",
       title = "Reads") +
  theme(axis.text = element_text(size = 8)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(strip.text = element_text(size = 7))

p.scatter.allv.euk.species.comp

ggsave(p.scatter.allv.euk.species.comp, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_unclassified_Viruses_family_euk_scatterplot_comp.pdf', width = 7, height = 6, dpi = 600)
  
# Look at unclassified Viruses Eukaryotic species, dark matter - require at least 10 reads HOLD
dark.euk.species <- c("unclassified Viruses species", "uncultured human fecal virus", "uncultured marine virus", "uncultured virus", "unidentified virus")

df.virus.fin.reads.dark.species.comp <- df.virus.fin.reads.30 %>%
  group_by(family,genus, species)  %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species %in% dark.euk.species) %>%
  droplevels() %>%
  mutate(Count = n())

# Plot
p.scatter.allv.euk.dark.comp <- ggplot(df.virus.fin.reads.dark.species.comp, aes(x = alnlen, y = pident, color = alnType)) +
  geom_jitter(position = position_jitter(width = 0.05, height = 0.05), size = 0.5, alpha = 0.5) +
  facet_wrap(~species) +
  geom_hline(aes(yintercept = 70), lty = 2, color = "gray") + # strange ggplot2 behaviour. Have to add this as an aes() if using fct_reorder
  geom_vline(aes(xintercept = 150), lty = 2, color = "gray") +
  labs(x = "Alignment Length",
       y = "% ID",
       color = "Query Type",
       caption = "%ID and alignment length for unclassified Viruses family; unknown eukaryotic viruses.\n Dashed lines indicate 70% ID (y-axis) and 150 bases amino acids (x-axis)",
       title = "Reads") +
  theme(axis.text = element_text(size = 8)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(strip.text = element_text(size = 7))

p.scatter.allv.euk.dark.comp

ggsave(p.scatter.allv.euk.dark.comp, filename = './lung_only/figures/reads/2023_01_05_RC2_IBD_f3_virome_reads_unclassified_Viruses_family_dark_scatterplot_comp.pdf', width = 7, height = 5, dpi = 600)

```

# Plot families of interest
```{r plot-families-of-interest}
# Coronaviridae
df.virus.fin.reads.corona <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Coronaviridae") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.corona <- ggplot(df.virus.fin.reads.corona, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Coronaviridae", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.corona

# Plot
ggsave(p.df.virus.fin.reads.corona, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_corona_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
corona.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Coronaviridae" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

corona.virus.pro$targetName = swr(corona.virus.pro$targetName)

p.corona.virus.pro  <- ggplot(corona.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Coronaviridae", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.corona.virus.pro

ggsave(p.corona.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_corona_protein_map.pdf', width = 9, height = 13, dpi = 600)

table(corona.virus.pro$targetID)
# A0A096XNG9 A0A0A0YRB5 A0A0B4N6G8 A0A0B4N7I9 A0A0B4Q809 A0A0G2RC85 A0A0G2RCC3 A0A0K0L9K1 A0A0K2RW50 A0A0U2GN94 
#          1          2         10          2          2         14         16          1          8         18 
# A0A127AT05 A0A140E067 A0A1D6X7X2 A0A1S5VGE8 A0A1S6KXQ5 A0A1V0IG06 A0A1W6DZR0 A0A1W6DZW1 A0A1Y0EV31 A0A218NKX9 
#         31          2          1         14          7          2          5          2          3          1 
# A0A220F1Q3 A0A2D1QW21 A0A2H4Z084 A0A2H4ZBM7 A0A2P1IQC5 A0A2Z4HFJ9 A0A2Z4HFK5 A0A2Z4HFU6 A0A3G3NHM9 A0A3Q9R4C4 
#          1         23         64          1          1          7         95          1          1          1 
# A0A3T0IBK1 A0A482KBQ0 A0A513Q8G6 A0A513Q8I6 A0A5B9MRZ0 A0A5B9MVH8 A0A5J6NRM4 A0A679DWY4 A0A679DXP4     H9AA59 
#         10        179         29          1          6          1          2          1          1          1 
#     H9AA60     P0C6U7     Q2QKN6     Q5ICX5     Q6TY37     T1PZ33     U3M761     U3M7L4     U3M820     V5MYI3 
#          1        561          2          1          1          1          1          1          2          2 
#     V5MYJ9     V5MYR7     V5MZ93     V5N116 
#          2          2          1          2 

# Viruses > Riboviria > Orthornavirae > Pisuviricota > Pisoniviricetes > Nidovirales > Cornidovirineae > Coronaviridae > Orthocoronavirinae > Betacoronavirus > Embecovirus > Betacoronavirus 1
# P0C6U7 (561) Human coronavirus OC43 Replicase polyprotein 1a
# A0A482KBQ0 (179) Human coronavirus OC43 ORF1a
# A0A2Z4HFK5 (95) Human coronavirus OC43 Replicase polyprotein 1ab
# A0A2H4Z084 (64) Porcine hemagglutinating encephalomyelitis virus Orf1a polyprotein

# Look at nt hits
corona.virus.nt <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Coronaviridae" ) %>%
  filter(alnType =="nt") %>%
  droplevels() 

table(corona.virus.nt$targetID,corona.virus.nt$pident)
                       
  #                       74.5 83.5 85 85.4 87.8 89.4 93.4 95.6 97.2 99.1 100
  # tid|31631|KF512574.1     0    0  0    1    0    0    0    0    0    0   0  Human coronavirus OC43 
  # tid|31631|KF923897.1     0    1  0    0    0    0    0    0    0    0   0  Human coronavirus OC43 
  # tid|31631|KF923922.1     0    0  0    0    0    0    0    0    1    0   0  Human coronavirus OC43 
  # tid|31631|LC315647.1     0    0  0    0    1    0    0    0    0    0   0  Human coronavirus OC43 
  # tid|31631|MG197713.1     1    0  0    0    0    0    0    0    0    0   0  Human coronavirus OC43 
  # tid|31631|MK303623.1     0    0  1    0    0    0    1    1    0    0   0 ** Human coronavirus OC43 
  # tid|31631|MK303625.1     0    0  0    0    0    1    0    0    0    0   0  Human coronavirus OC43 
  # tid|694009|CS480537.1    0    0  0    0    0    0    0    0    0    1   3 ** Severe acute respiratory syndrome-related coronavirus; 37,971 bp genomic sequence

# CoV OC43 YAH
cov.genome.map <- fread(file='./lung_only/data/corona_map/FFM_lung_only_all_reads_corona_genomecov.txt', sep='\t', header=TRUE,   colClasses = c("chromosome"  = "character", "position" = "numeric", "depth" = "numeric", "region" = "character"))

mean(cov.genome.map$depth)
# [1] 7.119496

# Color by Corona genome feature
corona.cols  <- c("non_coding" = "#000000", "orf1ab" = "#FF0000", "intergenic" = "#9FB6CD", "NS2a" = "#FF7F00", "HE" = "#B8860B", "S" = "#228B22", "NS5a" = "#00B2EE", "E" = "#0000FF", "M" = "#8A2BE2", "N" = "#8B008B")

# "non_coding" = "black" = #000000, "orf1ab" = "red" =  #FF0000, "intergenic" = "slategray3" = #9FB6CD, "NS2a" = "darkorange1" = #FF7F00, "HE" = "darkgoldenrod1", "S" = "forestgreen" = #228B22, "NS5a" = "deepskyblue2" = #00B2EE, "E" = "blue" = #0000FF, "M" = "blueviolet" = #8A2BE2, "N" = "darkmagenta" = #8B008B

cov.genome.map$region <- factor(cov.genome.map$region, levels = c("non_coding", "orf1ab", "intergenic", "NS2a", "HE", "S", "NS5a", "E", "M", "N"))

p.corona.genome.depth  <- ggplot(cov.genome.map, aes(x = position, y = (depth+0.00001), color = region)) +
  geom_point(aes(x = position, y = depth)) + # , alpha = 0.2
  theme(panel.background = element_blank(), panel.grid.minor = element_blank(), panel.grid.major = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  scale_y_continuous(trans='log10') +
  labs(title="Coronaviridae OC43", y = "log10 Depth", x = "Position") +
  scale_color_manual(values = corona.cols, name = "Genome region") +
  theme(legend.key = element_rect(colour = NA, fill = NA))

p.corona.genome.depth

ggsave(p.corona.genome.depth, filename = './lung_only/figures/reads/2023_02_03_formalin_fixed_measles_lung_only_Coronaviridae_OC43_read_depth.pdf', width = 6, height = 5, dpi = 600)

p.corona.oc43.genome <- ggplot() + 
  xlim(1, 30704) +
  ylim(25,35) +
  geom_hline(yintercept=30, linetype="solid", color = "black")  + 
  geom_rect(aes(xmin = 186, xmax = 21472, ymin = 25, ymax = 35), fill = "#FF0000") +
  geom_text(aes(x = 10829, y = 30, label = "orf1ab")) +
  geom_rect(aes(xmin = 21482, xmax = 22318, ymin = 25, ymax = 35), fill = "#FF7F00") +
  geom_text(aes(x = 21870, y = 32, label = "NS2a")) +
  geom_rect(aes(xmin = 22330, xmax = 23613, ymin = 25, ymax = 35), fill = "#B8860B") +
  geom_text(aes(x = 22972, y = 30, label = "HE")) +
  geom_rect(aes(xmin = 23628, xmax = 27707, ymin = 25, ymax = 35), fill = "#228B22") +
  geom_text(aes(x = 25668, y = 32, label = "S"),color="white") +
  geom_rect(aes(xmin = 27783, xmax = 28112, ymin = 25, ymax = 35), fill = "#00B2EE") +
  geom_text(aes(x = 27860, y = 30, label = "NS5a"),color="white") +
  geom_rect(aes(xmin = 28099, xmax = 28353, ymin = 25, ymax = 35), fill = "#0000FF") +
  geom_text(aes(x = 28226, y = 32, label = "E"),color="white") +
  geom_rect(aes(xmin = 28368, xmax = 29060, ymin = 25, ymax = 35), fill = "#8A2BE2") +
  geom_text(aes(x = 28714, y = 30, label = "M"),color="white") +
  geom_rect(aes(xmin = 29070, xmax = 30416, ymin = 25, ymax = 35), fill = "#8B008B") +
  geom_text(aes(x = 29743, y = 32, label = "N"),color="white") +
  theme(legend.position="none") +
  theme(panel.background = element_blank(), panel.grid.minor = element_blank(), panel.grid.major = element_blank()) +
  theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) +
  theme(axis.title.y=element_blank(), axis.text.y=element_blank(), axis.ticks.y=element_blank()) +
  ggtitle("Human coronavirus OC43 isolate TNP 1264")

p.corona.oc43.genome

ggsave(p.corona.oc43.genome, filename = './lung_only/figures/reads/2023_02_03_formalin_fixed_measles_lung_only_Coronaviridae_OC43_genome.pdf', width = 6, height = 2, dpi = 600)

p.corona.oc43.depth <- plot_grid(p.corona.genome.depth, p.corona.oc43.genome, ncol=1,  rel_heights = c(2, 1))

ggsave(p.corona.oc43.depth, filename = './lung_only/figures/reads/2023_02_03_formalin_fixed_measles_lung_only_Coronaviridae_OC43_reads_genome_depth.pdf', width = 6, height = 6, dpi = 600)

# Paramyxoviridae
df.virus.fin.reads.para <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Paramyxoviridae") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.para <- ggplot(df.virus.fin.reads.para, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Paramyxoviridae", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.para

# Plot
ggsave(p.df.virus.fin.reads.para, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_para_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
para.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Paramyxoviridae" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

para.virus.pro$targetName = swr(para.virus.pro$targetName)

p.para.virus.pro  <- ggplot(para.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Paramyxoviridae", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.para.virus.pro

ggsave(p.para.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_para_protein_map.pdf', width = 10, height = 15, dpi = 600)

# Map reads to proteins
measles.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Paramyxoviridae" ) %>%
  filter(species == "Measles morbillivirus") %>%
  filter(alnType =="aa") %>%
  droplevels() 

para.virus.pro$targetName = swr(para.virus.pro$targetName)

p.measles.virus.pro  <- ggplot(measles.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Measles morbillivirus", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.measles.virus.pro

ggsave(p.measles.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_measles_protein_map.pdf', width = 11, height = 16, dpi = 600)

# Adenoviridae
df.virus.fin.reads.adeno <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Adenoviridae") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.adeno <- ggplot(df.virus.fin.reads.adeno, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Adenoviridae", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.adeno

# Plot
ggsave(p.df.virus.fin.reads.adeno, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_adeno_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
adeno.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Adenoviridae" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

adeno.virus.pro$targetName = swr(adeno.virus.pro$targetName)

p.adeno.virus.pro  <- ggplot(adeno.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Adenoviridae", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.adeno.virus.pro

ggsave(p.adeno.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_adeno_protein_map.pdf', width = 9, height = 13, dpi = 600)

# Birnaviridae
df.virus.fin.reads.birna <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Birnaviridae") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.birna <- ggplot(df.virus.fin.reads.birna, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Birnaviridae", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.birna

# Plot
ggsave(p.df.virus.fin.reads.birna, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_birna_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
birna.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Birnaviridae" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

birna.virus.pro$targetName = swr(birna.virus.pro$targetName)

p.birna.virus.pro  <- ggplot(birna.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Birnaviridae", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.birna.virus.pro

ggsave(p.birna.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_birna_protein_map.pdf', width = 6, height = 4, dpi = 600)

# 1 aa Drosphila melanogaster birnavirus read 54% ID - FP

# Herpesviridae 
df.virus.fin.reads.herpes <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Herpesviridae") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.herpes <- ggplot(df.virus.fin.reads.herpes, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Herpesviridae", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.herpes

# Plot
ggsave(p.df.virus.fin.reads.herpes, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_herpes_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
herpes.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Herpesviridae" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

herpes.virus.pro$targetName = swr(herpes.virus.pro$targetName)

p.herpes.virus.pro  <- ggplot(herpes.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Herpesviridae", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.herpes.virus.pro

ggsave(p.herpes.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_herpes_protein_map.pdf', width = 5, height = 5, dpi = 600)

# Orthomyxoviridae 
df.virus.fin.reads.ortho <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Orthomyxoviridae") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.ortho <- ggplot(df.virus.fin.reads.ortho, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Orthomyxoviridae", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.ortho

# Plot
ggsave(p.df.virus.fin.reads.ortho, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_ortho_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
ortho.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Orthomyxoviridae" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

ortho.virus.pro$targetName = swr(ortho.virus.pro$targetName)

p.ortho.virus.pro  <- ggplot(ortho.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Orthomyxoviridae", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.ortho.virus.pro

ggsave(p.ortho.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_ortho_protein_map.pdf', width = 5, height = 5, dpi = 600)

# Mapping
flu.blast <- as_tibble(fread(file = "./lung_only/data/ortho_map/Influenza_A_blastn_results.tsv", header = TRUE, stringsAsFactors = TRUE))

flu.blast.np <- flu.blast %>%
  filter(Scientific_Name == "Influenza A virus A/Brevig Mission/1/1918(H1N1)) nucleoprotein") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- flu.blast.np$Acc_Len + 100

p.flu.np.map  <- ggplot(flu.blast.np, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Influenza A virus A/Brevig Mission/1/1918(H1N1): nucleoprotein") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.flu.np.map

ggsave(p.flu.np.map, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_flu_np_map.pdf', width = 8, height = 4, dpi = 600)

# Polymerase PB1
flu.blast.pb <- flu.blast %>%
  filter(Scientific_Name == "Influenza A virus A/Brevig Mission/1/1918(H1N1)) polymerase PB1") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- flu.blast.pb$Acc_Len + 100

p.flu.pb.map  <- ggplot(flu.blast.pb, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Influenza A virus A/Brevig Mission/1/1918(H1N1): polymerase PB1") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.flu.pb.map

ggsave(p.flu.pb.map, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_flu_pb1_map.pdf', width = 8, height = 4, dpi = 600)

# NT blast
p.flu.blast <- ggplot(flu.blast, aes(x = alnlen, y = Per_Ident, color = alnType)) +
  xlim(0, 350) +
  geom_jitter(position = position_jitter(width = 0.05, height = 0.05), size = 1.0, alpha = 0.5) +
  facet_wrap(~swr(Scientific_Name)) +
  geom_hline(aes(yintercept = 90), lty = 2, color = "gray") + # strange ggplot2 behaviour. Have to add this as an aes() if using fct_reorder
  geom_vline(aes(xintercept = 150), lty = 2, color = "gray") +
  labs(x = "Alignment Length",
       y = "% ID",
       caption = "%ID and nucleotide alignment length for all detected viral families.\n Dashed lines indicate 90% ID (y-axis) and 150 bases nucleotides (x-axis)") +
  theme(axis.text = element_text(size = 8)) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(strip.text = element_text(size = 7))

p.flu.blast

ggsave(p.flu.blast, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_flu_nt_blast.pdf', width = 5, height = 3.5, dpi = 600)


# Picobirnaviridae
df.virus.fin.reads.picobirna <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Picobirnaviridae") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.picobirna <- ggplot(df.virus.fin.reads.picobirna, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Picobirnaviridae", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.picobirna

# Plot
ggsave(p.df.virus.fin.reads.picobirna, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_picobirna_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
picobirna.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Picobirnaviridae" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

picobirna.virus.pro$targetName = swr(picobirna.virus.pro$targetName)

p.picobirna.virus.pro  <- ggplot(picobirna.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Picobirnaviridae", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.picobirna.virus.pro

ggsave(p.picobirna.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_picobirna_protein_map.pdf', width = 6, height = 8, dpi = 600)

# Pneumoviridae
df.virus.fin.reads.pneumo <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Pneumoviridae") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.pneumo <- ggplot(df.virus.fin.reads.pneumo, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Pneumoviridae", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.pneumo

# Plot
ggsave(p.df.virus.fin.reads.pneumo, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_pneumo_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
pneumo.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Pneumoviridae" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

pneumo.virus.pro$targetName = swr(pneumo.virus.pro$targetName)

p.pneumo.virus.pro  <- ggplot(pneumo.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Pneumoviridae", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.pneumo.virus.pro

ggsave(p.pneumo.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_pneumo_protein_map.pdf', width = 9, height = 13, dpi = 600)

# Human orthopneumovirus read mapping
pneumoviridae.blast <- as_tibble(fread(file = "./lung_only/data/pneumo_map/Pneumoviridae_virus_blastn.tsv", header = TRUE, stringsAsFactors = TRUE))

# Human orthopneumovirus isolate A-TX-012b-2015-WGS
orthopneumovirus.blast <- pneumoviridae.blast %>%
  filter(Scientific_Name == "Human orthopneumovirus") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- orthopneumovirus.blast$Acc_Len + 100

p.orthopneumovirus.map  <- ggplot(orthopneumovirus.blast, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  ylim(0,16) +
  geom_rect(aes(xmin = 55, xmax = 474, ymin = 11.5, ymax = 12.5), fill = "#FF0000") +
  geom_text(aes(x = 264, y = 12.2, label = "NS1")) +
  geom_rect(aes(xmin = 584, xmax = 958, ymin = 11.5, ymax = 12.5), fill = "#FF0000") +
  geom_text(aes(x = 771, y = 11.8, label = "NS2")) +
  geom_rect(aes(xmin = 1096, xmax = 2271, ymin = 11.5, ymax = 12.5), fill = "#FF7F00") +
  geom_text(aes(x = 1684, y = 12.2, label = "N")) +
  geom_rect(aes(xmin = 2303, xmax = 3028, ymin = 11.5, ymax = 12.5), fill = "#B8860B") +
  geom_text(aes(x = 2666, y = 12.2, label = "P")) +
  geom_rect(aes(xmin = 3211, xmax = 3981, ymin = 11.5, ymax = 12.5), fill = "#228B22") +
  geom_text(aes(x = 3596, y = 12.2, label = "M"),color="white") +
  geom_rect(aes(xmin = 4251, xmax = 4445, ymin = 11.5, ymax = 12.5), fill = "#00B2EE") +
  geom_text(aes(x = 4348, y = 12.2, label = "SH"),color="black") +
  geom_rect(aes(xmin = 4636, xmax = 5601, ymin = 11.5, ymax = 12.5), fill = "#0000FF") +
  geom_text(aes(x = 5119, y = 12.2, label = "G"),color="white") +
  geom_rect(aes(xmin = 5681, xmax = 7405, ymin = 11.5, ymax = 12.5), fill = "#8A2BE2") +
  geom_text(aes(x = 6543, y = 12.2, label = "F"),color="white") +
  geom_rect(aes(xmin = 7626, xmax = 8210, ymin = 11.5, ymax = 12.5), fill = "#8B008B") +
  geom_text(aes(x = 7918, y = 12.2, label = "M2"),color="white") +
  geom_rect(aes(xmin = 8185, xmax = 8451, ymin = 11.5, ymax = 12.5), fill = "#8B008B") +
  geom_text(aes(x = 8318, y = 11.8, label = "M2"),color="white") +
  geom_rect(aes(xmin = 8518, xmax = 15015, ymin = 11.5, ymax = 12.5), fill = "#000000") +
  geom_text(aes(x = 11767, y = 12.2, label = "L"),color="white") +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Human orthopneumovirus isolate A-TX-012b-2015-WGS") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")

p.orthopneumovirus.map

ggsave(p.orthopneumovirus.map, filename = './lung_only/figures/reads/2023_02_08_formalin_fixed_measles_lung_only_orthopneumovirus_map.pdf', width = 10, height = 5, dpi = 600)

# Human metapneumovirus isolate NL/13/06/B2
metapneumovirus.blast <- pneumoviridae.blast %>%
  filter(Scientific_Name == "Human metapneumovirus") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- metapneumovirus.blast$Acc_Len + 100

p.metapneumovirus.map  <- ggplot(metapneumovirus.blast, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  ylim(0,3) +
  geom_rect(aes(xmin = 54, xmax = 1238, ymin = 1.9, ymax = 2.1), fill = "#FF7F00") +
  geom_text(aes(x = 646, y = 2.05, label = "N")) +
  geom_rect(aes(xmin = 1262, xmax = 2146, ymin = 1.9, ymax = 2.1), fill = "#B8860B") +
  geom_text(aes(x = 1704, y = 2.05, label = "P")) +
  geom_rect(aes(xmin = 2164, xmax = 2943, ymin = 1.9, ymax = 2.1), fill = "#228B22") +
  geom_text(aes(x = 2554, y = 2.05, label = "M"),color="white") +
  geom_rect(aes(xmin = 5468, xmax = 6001, ymin = 1.9, ymax = 2.1), fill = "#00B2EE") +
  geom_text(aes(x = 5735, y = 2.05, label = "SH"),color="black") +
  geom_rect(aes(xmin = 6197, xmax = 6922, ymin = 1.9, ymax = 2.1), fill = "#0000FF") +
  geom_text(aes(x = 6560, y = 2.05, label = "G"),color="white") +
  geom_rect(aes(xmin = 3065, xmax = 4684, ymin = 1.9, ymax = 2.1), fill = "#8A2BE2") +
  geom_text(aes(x = 3875, y = 2.05, label = "F"),color="white") +
  geom_rect(aes(xmin = 4711, xmax = 5274, ymin = 1.9, ymax = 2.1), fill = "#8B008B") +
  geom_text(aes(x = 4993, y = 1.98, label = "M2.1"),color="white") +
  geom_rect(aes(xmin = 5222, xmax = 6001, ymin = 1.9, ymax = 2.1), fill = "#8B008B") +
  geom_text(aes(x = 5612, y = 2.05, label = "M2-2"),color="white") +
  geom_rect(aes(xmin = 7115, xmax = 13132, ymin = 1.9, ymax = 2.1), fill = "black") +
  geom_text(aes(x = 10124, y = 2.05, label = "L"),color="white") +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Human metapneumovirus isolate NL/13/06/B2") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.metapneumovirus.map

ggsave(p.metapneumovirus.map, filename = './lung_only/figures/reads/2023_02_08_formalin_fixed_measles_lung_only_metapneumovirus_map.pdf', width = 9, height = 4, dpi = 600)

# Reoviridae
df.virus.fin.reads.reo <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Reoviridae") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.reo <- ggplot(df.virus.fin.reads.reo, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Reoviridae", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.reo

# Plot
ggsave(p.df.virus.fin.reads.reo, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_reo_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
reo.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Reoviridae" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

reo.virus.pro$targetName = swr(reo.virus.pro$targetName)

p.reo.virus.pro  <- ggplot(reo.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Reoviridae", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.reo.virus.pro

ggsave(p.reo.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_reo_protein_map.pdf', width = 9, height = 13, dpi = 600)

# -  Kadipiro virus in Denmark (bat feces) [located while looking for COVID] Viruses 2021, 13(6), 1073; https://doi.org/10.3390/v13061073 

table(reo.virus.pro$targetID)
# A0A0K1YWL2 A0A1I7PCM6 A0A1I7PCM7 A0A1L3KP15 A0A1L3KP19 A0A1L3KP22 A0A1L3KP40 A0A1L3KPA2 A0A343WUA3 A0A343WUA7 
#          1          2          4          2          3          2          1          1          3          2 
#     Q9INI8     Q9INJ2     Q9YWP7 
#         12          2          1 

# Q9INI8: Kadipiro virus (strain: JKT-7075) 
# A0A1I7PCM7: Kadipiro virus isolate P3 VP2 gene, partial cds.
# A0A343WUA3: Kadipiro virus strain SDKL1625 segment 2 VP2 gene, complete cds.

# Kadipiro virus read mapping
kadipiro.blast <- as_tibble(fread(file = "./lung_only/data/reo_fish/kadipiro_map/Kadipiro_virus_blastn.tsv", header = TRUE, stringsAsFactors = TRUE))

# Kadipiro virus rat 2019 VP1 RdRp partial
kadipiro.blast.VP1 <- kadipiro.blast %>%
  filter(Scientific_Name == "Kadipiro virus rat 2019 VP1 RdRp partial") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- kadipiro.blast.VP1$Acc_Len + 100

p.kadipiro.VP1.map  <- ggplot(kadipiro.blast.VP1, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Kadipiro virus rat 2019 VP1 RdRp partial") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.kadipiro.VP1.map

ggsave(p.kadipiro.VP1.map, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_kadipiro_VP1_map.pdf', width = 8, height = 4, dpi = 600)

# Kadipiro virus rat 2019 VP2
kadipiro.blast.VP2 <- kadipiro.blast %>%
  filter(Scientific_Name == "Kadipiro virus rat 2019 VP2") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- kadipiro.blast.VP2$Acc_Len + 100

p.kadipiro.VP2.map  <- ggplot(kadipiro.blast.VP2, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Kadipiro virus rat 2019 VP2") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.kadipiro.VP2.map

ggsave(p.kadipiro.VP2.map, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_kadipiro_VP2_map.pdf', width = 8, height = 4, dpi = 600)

# Kadipiro virus rat 2019 VP3 partial
kadipiro.blast.VP3 <- kadipiro.blast %>%
  filter(Scientific_Name == "Kadipiro virus rat 2019 VP3 partial") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- kadipiro.blast.VP3$Acc_Len + 100

p.kadipiro.VP3.map  <- ggplot(kadipiro.blast.VP3, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Kadipiro virus rat 2019 VP3 partial") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.kadipiro.VP3.map

ggsave(p.kadipiro.VP3.map, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_kadipiro_VP3_map.pdf', width = 8, height = 4, dpi = 600)

# Kadipiro virus rat 2019 VP4
kadipiro.blast.VP4 <- kadipiro.blast %>%
  filter(Scientific_Name == "Kadipiro virus rat 2019 VP4") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- kadipiro.blast.VP4$Acc_Len + 100

p.kadipiro.VP4.map  <- ggplot(kadipiro.blast.VP4, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Kadipiro virus rat 2019 VP4") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.kadipiro.VP4.map

ggsave(p.kadipiro.VP4.map, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_kadipiro_VP4_map.pdf', width = 8, height = 4, dpi = 600)

# Kadipiro virus rat 2019 VP5
kadipiro.blast.VP5 <- kadipiro.blast %>%
  filter(Scientific_Name == "Kadipiro virus rat 2019 VP5") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- kadipiro.blast.VP5$Acc_Len + 100

p.kadipiro.VP5.map  <- ggplot(kadipiro.blast.VP5, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Kadipiro virus rat 2019 VP5") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.kadipiro.VP5.map

ggsave(p.kadipiro.VP5.map, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_kadipiro_VP5_map.pdf', width = 8, height = 4, dpi = 600)

# Kadipiro virus rat 2019 VP6
kadipiro.blast.VP6 <- kadipiro.blast %>%
  filter(Scientific_Name == "Kadipiro virus rat 2019 VP6") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- kadipiro.blast.VP6$Acc_Len + 100

p.kadipiro.VP6.map  <- ggplot(kadipiro.blast.VP6, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Kadipiro virus rat 2019 VP6") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.kadipiro.VP6.map

ggsave(p.kadipiro.VP6.map, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_kadipiro_VP6_map.pdf', width = 8, height = 4, dpi = 600)

# Kadipiro virus rat 2019 VP7
kadipiro.blast.VP7 <- kadipiro.blast %>%
  filter(Scientific_Name == "Kadipiro virus rat 2019 VP7") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- kadipiro.blast.VP7$Acc_Len + 100

p.kadipiro.VP7.map  <- ggplot(kadipiro.blast.VP7, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Kadipiro virus rat 2019 VP7") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.kadipiro.VP7.map

ggsave(p.kadipiro.VP7.map, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_kadipiro_VP7_map.pdf', width = 8, height = 4, dpi = 600)

# Kadipiro virus rat 2019 VP8
kadipiro.blast.VP8 <- kadipiro.blast %>%
  filter(Scientific_Name == "Kadipiro virus rat 2019 VP8") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- kadipiro.blast.VP8$Acc_Len + 100

p.kadipiro.VP8.map  <- ggplot(kadipiro.blast.VP8, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Kadipiro virus rat 2019 VP8") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.kadipiro.VP8.map

ggsave(p.kadipiro.VP8.map, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_kadipiro_VP8_map.pdf', width = 8, height = 4, dpi = 600)

# Kadipiro virus rat 2019 VP9
kadipiro.blast.VP9 <- kadipiro.blast %>%
  filter(Scientific_Name == "Kadipiro virus rat 2019 VP9") %>%
  mutate(qstart_fin = ifelse((qstart - qend > 0), qend, qstart)) %>%
  mutate(qend_fin = ifelse((qstart - qend > 0), qstart, qend)) %>%
  mutate(seq_start = ifelse((tstart - qstart_fin > 0), (tstart - qstart_fin), 0)) %>%
  mutate(seq_end = (tend + (Length -alnlen))) %>%
  droplevels()

x_right <- kadipiro.blast.VP9$Acc_Len + 100

p.kadipiro.VP9.map  <- ggplot(kadipiro.blast.VP9, aes(x = Acc_Len, y = seqID)) +
  xlim(0, x_right[1]) +
  geom_segment(aes(x = tstart, xend = tend, y = seqID, yend = seqID, color="aligned")) +
  geom_point(aes(x = tstart, y = seqID), color="red") +
  geom_point(aes(x = tend, y = seqID), color="red") +
  geom_point(alpha = 0.5, aes(x = seq_start, y = seqID), color = "gray") +
  geom_point(alpha = 0.5, aes(x = seq_end, y = seqID), color = "gray") +   
  geom_segment(alpha = 0.5, aes(x = seq_start, xend = seq_end, y = seqID, yend = seqID, color = "unaligned"), lty = 5)  +
  scale_y_discrete(position = "right") +
  scale_color_manual(name = "Alignment:",
                          breaks = c("unaligned", "aligned"),
                          values = c("unaligned" = "grey", "aligned" = "red")) +
  theme(panel.background = element_rect(fill = "white", colour = "black")) +
  theme(legend.position = "top") +
  labs(title="Kadipiro virus rat 2019 VP9") +
  geom_segment(alpha = 0.4, aes(x = 1, xend = Acc_Len, species, yend = species), color = "black") +
  scale_y_discrete(position = "right") +
  labs(y = "", x = "Position")  +
  theme(panel.background = element_rect(fill = "white", colour = "black")) 

p.kadipiro.VP9.map

ggsave(p.kadipiro.VP9.map, filename = './lung_only/figures/reads/2023_02_06_formalin_fixed_measles_lung_only_kadipiro_VP9_map.pdf', width = 8, height = 4, dpi = 600)

# Retroviridae
df.virus.fin.reads.retro <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "Retroviridae") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.retro <- ggplot(df.virus.fin.reads.retro, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Retroviridae", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.retro

# Plot
ggsave(p.df.virus.fin.reads.retro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_retro_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
retro.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "Retroviridae" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

retro.virus.pro$targetName = swr(retro.virus.pro$targetName)

p.retro.virus.pro  <- ggplot(retro.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Retroviridae", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.retro.virus.pro

ggsave(p.retro.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_retro_protein_map.pdf', width = 9, height = 13, dpi = 600)

# HIV??? - most likely false positive

# unclassified Picornavirales family
df.virus.fin.reads.upf <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "unclassified Picornavirales family") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.upf <- ggplot(df.virus.fin.reads.upf, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="unclassified Picornavirales family", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.upf

# Plot
ggsave(p.df.virus.fin.reads.upf, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_upf_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
upf.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "unclassified Picornavirales family" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

upf.virus.pro$targetName = swr(upf.virus.pro$targetName)

p.upf.virus.pro  <- ggplot(upf.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="unclassified Picornavirales family", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.upf.virus.pro

ggsave(p.upf.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_upf_protein_map.pdf', width = 9, height = 13, dpi = 600)

```

# Plot unclassified viruses
```{r plot-unclassified-viruses}
# unclassified Viruses family
df.virus.fin.reads.uvf <- df.virus.fin.reads %>%
  group_by(family)  %>%
  filter(family == "unclassified Viruses family") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.uvf <- ggplot(df.virus.fin.reads.uvf, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="unclassified Viruses family", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.uvf

# Plot
ggsave(p.df.virus.fin.reads.uvf, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
uvf.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(alnType =="aa") %>%
  droplevels() 

uvf.virus.pro$targetName = swr(uvf.virus.pro$targetName)

p.uvf.virus.pro  <- ggplot(uvf.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="unclassified Viruses family", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.uvf.virus.pro

ggsave(p.uvf.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_protein_map.pdf', width = 11, height = 15, dpi = 600)

desired.picorna.species <- c("Beihai picorna-like virus 14", "Beihai picorna-like virus 15", "Beihai picorna-like virus 17", "Beihai picorna-like virus 20", "Beihai picorna-like virus 21", "Beihai picorna-like virus 37", "Beihai picorna-like virus 4", "Beihai picorna-like virus 53", "Sanxia picorna-like virus 3 ", "Sanxia picorna-like virus 7", "Shahe picorna-like virus 7", "Wenzhou picorna-like virus 5", "Wenzhou picorna-like virus 6", "Wenzhou picorna-like virus 7")

df.virus.fin.reads.picorna <- df.virus.fin.reads %>%
  group_by(family,genus, species)  %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species %in% desired.picorna.species) %>%
  droplevels() %>%
  mutate(Count = n()) 

# unclassified Viruses family Picorna-like
df.virus.fin.reads.picorna <- df.virus.fin.reads.picorna %>%
  group_by(family)  %>%
  filter(family == "unclassified Viruses family") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.picorna <- ggplot(df.virus.fin.reads.picorna, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Picorna-like virus", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.picorna

# Plot
ggsave(p.df.virus.fin.reads.picorna, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_picorna_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
uvf.picorna.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species %in% desired.picorna.species) %>%
  filter(alnType =="aa") %>%
  droplevels() 

uvf.picorna.virus.pro$targetName = swr(uvf.picorna.virus.pro$targetName)

p.uvf.picorna.virus.pro  <- ggplot(uvf.picorna.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Picorna-like virus", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.uvf.picorna.virus.pro

ggsave(p.uvf.picorna.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_picorna_protein_map.pdf', width = 9, height = 13, dpi = 600)

# Ellsworth virus
df.virus.fin.reads.ells <- df.virus.fin.reads %>%
  group_by(family,genus, species)  %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species == "Ellsworth virus") %>%
  droplevels() %>%
  mutate(Count = n()) 

df.virus.fin.reads.ells <- df.virus.fin.reads.ells %>%
  group_by(family)  %>%
  filter(family == "unclassified Viruses family") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.ells <- ggplot(df.virus.fin.reads.ells, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Ellsworth virus", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.ells

# Plot
ggsave(p.df.virus.fin.reads.ells, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_ells_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
uvf.ells.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species == "Ellsworth virus") %>%
  filter(alnType =="aa") %>%
  droplevels() 

uvf.ells.virus.pro$targetName = swr(uvf.ells.virus.pro$targetName)

p.uvf.ells.virus.pro  <- ggplot(uvf.ells.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Ellsworth virus", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.uvf.ells.virus.pro

ggsave(p.uvf.ells.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_ells_protein_map.pdf', width = 4, height = 5, dpi = 600)

# Riiser virus
df.virus.fin.reads.riis <- df.virus.fin.reads %>%
  group_by(family,genus, species)  %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species == "Riiser virus") %>%
  droplevels() %>%
  mutate(Count = n()) 

df.virus.fin.reads.riis <- df.virus.fin.reads.riis %>%
  group_by(family)  %>%
  filter(family == "unclassified Viruses family") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.riis <- ggplot(df.virus.fin.reads.riis, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Riiser virus", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.riis

# Plot
ggsave(p.df.virus.fin.reads.riis, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_riis_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
uvf.riis.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species == "Riiser virus") %>%
  filter(alnType =="aa") %>%
  droplevels() 

uvf.riis.virus.pro$targetName = swr(uvf.riis.virus.pro$targetName)

p.uvf.riis.virus.pro  <- ggplot(uvf.riis.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Riiser virus", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.uvf.riis.virus.pro

ggsave(p.uvf.riis.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_riis_protein_map.pdf', width = 4, height = 5, dpi = 600)

# Sylvanvirus sp.
df.virus.fin.reads.syl <- df.virus.fin.reads %>%
  group_by(family,genus, species)  %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species == "Sylvanvirus sp.") %>%
  droplevels() %>%
  mutate(Count = n()) 

df.virus.fin.reads.syl <- df.virus.fin.reads.syl %>%
  group_by(family)  %>%
  filter(family == "unclassified Viruses family") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.syl <- ggplot(df.virus.fin.reads.syl, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Sylvanvirus sp.", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.syl

# Plot
ggsave(p.df.virus.fin.reads.syl, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_syl_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
uvf.syl.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species == "Sylvanvirus sp.") %>%
  filter(alnType =="aa") %>%
  droplevels() 

uvf.syl.virus.pro$targetName = swr(uvf.syl.virus.pro$targetName)

p.uvf.syl.virus.pro  <- ggplot(uvf.syl.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Sylvanvirus sp.", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.uvf.syl.virus.pro

ggsave(p.uvf.syl.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_syl_protein_map.pdf', width = 6, height = 8, dpi = 600)

# Tombus-like virus
desired.tombus <- c("Changjiang tombus-like virus 4",  "Wenzhou tombus-like virus 2 ", "Wenzhou tombus-like virus 3")

df.virus.fin.reads.tlv <- df.virus.fin.reads %>%
  group_by(family,genus, species)  %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species %in% desired.tombus) %>%
  droplevels() %>%
  mutate(Count = n()) 

df.virus.fin.reads.tlv <- df.virus.fin.reads.tlv %>%
  group_by(family)  %>%
  filter(family == "unclassified Viruses family") %>%
  droplevels() %>%
  group_by(database_ID, alnType) %>%
  summarise(count = n()) %>%
  ungroup() %>%
  complete(database_ID, alnType, fill = list(count = 0)) %>%
  group_by(database_ID, alnType) 

p.df.virus.fin.reads.tlv <- ggplot(df.virus.fin.reads.tlv, aes(x = as.character(database_ID), y = count, fill = alnType)) +
  geom_bar(stat = "identity", position="dodge") +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  theme(legend.position = "right") +
  theme(panel.background = element_rect(fill = 'white', colour = 'black')) +
  labs(title="Tombus-like virus", x = "Database ID", y = "# of reads") +
  scale_fill_manual(values = alnType.cols, name = "Aln Type")

p.df.virus.fin.reads.tlv

# Plot
ggsave(p.df.virus.fin.reads.tlv, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_tlv_sample.pdf', width = 5, height = 3.5, dpi = 600)

# Map reads to proteins
uvf.tlv.virus.pro <- df.virus.fin.reads %>%
  group_by(targetName) %>%
  filter(count >0) %>%
  filter(family == "unclassified Viruses family" ) %>%
  filter(species %in% desired.tombus) %>%
  filter(alnType =="aa") %>%
  droplevels() 

uvf.tlv.virus.pro$targetName = swr(uvf.tlv.virus.pro$targetName)

p.uvf.tlv.virus.pro  <- ggplot(uvf.tlv.virus.pro, aes(x = tstart, y = pident)) +
  geom_segment(alpha = 0.4, aes(x = tstart, xend = tend, y = pident, yend = pident, color = species)) +
  geom_point(aes(x = tstart, y = pident, color = species, size = count), alpha = 0.4) +
  geom_point(aes(x = tend, y = pident, color = species, size = count), alpha = 0.4) +
  facet_wrap(~targetName, scales = "free_x") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
  labs(title="Tombus-like virus", y = "% ID", x = "Target Position", color = "Species", size = "CPM")

p.uvf.tlv.virus.pro

ggsave(p.uvf.tlv.virus.pro, filename = './lung_only/figures/reads/2023_01_05_formalin_fixed_measles_lung_only_virome_reads_uvf_tlv_protein_map.pdf', width = 5, height = 5, dpi = 600)

```
