---
title: "Celiac_Data_subGroups_Explore"
output: html_document
date: '2025-04-23'
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library("dplyr")
library("tidyverse")

options(timeout = 300)
install.packages("https://cran.r-project.org/src/contrib/Archive/vroom/vroom_1.5.7.tar.gz", repos = NULL, type = "source")
library(vroom)
library(data.table)
remove.packages(c("tidyverse", "tibble", "tidyr", "readr", "purrr", "dplyr", "forcats", "lubridate"))
install.packages("tidyverse", type = "binary")
library(tidyverse)
library(data.table)
install.packages("reshape2")
library(reshape2)

options(timeout = 300)

if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("DESeq2")


if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

# Explicitly set the correct Bioconductor mirror
options(repos = BiocManager::repositories())
BiocManager::install("DESeq2")


BiocManager::install(
    "DESeq2",
    site_repository = "https://bioconductor.org/packages/release/bioc"
)


library(DESeq2)
library(data.table)
library(tidyverse)
library(limma)
library(edgeR)
library(dplyr)
library(tibble)
```


# functions
```{r}


filter_and_save_wide_data <- function(genesCoverm, 
                                      covered_fraction_threshold = 0.75, 
                                      output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/subgroups/",
                                      fileName) {
  
  
  # Set column names to match the original structure
  colnames(genesCoverm) <- c("sampleID", "ORFID", "RPKM", "ReadCount", "Variance", "Mean", "covered_fraction", "covered_bases")
  
  # Apply coverage filter
  genesCoverm2 <- genesCoverm %>%
    mutate(
      ReadCount_modified = ifelse(covered_fraction >= covered_fraction_threshold, ReadCount, 0),
      Changed = ifelse(ReadCount != ReadCount_modified, 1, 0)
    ) %>%
    mutate(ReadCount = ReadCount_modified) %>%
    select(-ReadCount_modified)
  
  # Count the number of changes
  num_changes <- sum(genesCoverm2$Changed)
  print(paste("Number of changes:", num_changes))
  
  # Convert to wide format
  wide_rpkm_genes <- reshape2::dcast(genesCoverm2, ORFID ~ sampleID, value.var = "ReadCount")
  colnames(wide_rpkm_genes) <- gsub("_stats", "", colnames(wide_rpkm_genes))
  
  # Filter out rows with all zeroes
  filtered_wide_rpkm_genes <- wide_rpkm_genes %>%
    filter(rowSums(across(where(is.numeric))) != 0)
  
  # Save the filtered wide format data
  rds_filename_wide <- paste0(fileName,"_wide_rpkm_genes_", covered_fraction_threshold * 100, "Cov.rds")
  
  saveRDS(filtered_wide_rpkm_genes, file = file.path(output_path, rds_filename_wide))
  
  return(filtered_wide_rpkm_genes)
}



```


# load metadata
```{r}

metadata <- read.csv("~/Handley Lab Dropbox/16S/Celiac/metadata/metadata_final_update_11262024.csv") %>%
            column_to_rownames("sampleName")

View(metadata)


```


# prepare total contig count table
```{r}

contig <- fread("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig/Coverm_concatenated.txt") %>% #,col_names = FALSE
          filter(!V1 %like% "NovaSeq_N983_I13413_39587_Celiac_Leonard_Stool_01_GEMM_106_12M") %>%
          mutate(V1 = gsub("NovaSeq_N966_|NovaSeq_N978_|NovaSeq_N979_|NovaSeq_N980_|NovaSeq_N981_|NovaSeq_N982_|NovaSeq_N983_|_stats","",V1)) %>%
          mutate(V1 = str_replace_all(V1,"Leonard_Human_stool_Celiac","Celiac_Leonard_Stool")) %>%
          mutate(V1 = sapply(strsplit(V1,"_Stool_|_Stool_Celiac_"),"[",2)) %>%
          mutate(V1 = ifelse(V1 %like% "_6Y",gsub("_6Y","_72M",V1),
                                           ifelse(V1 %like% "_7Y",gsub("_7Y","_84M",V1),V1)))


```


US
```{r}


############################ US ################################

metadata.US <- metadata %>%
               filter(Country == "USA")
# 205  24

table(metadata.US$Dx.Status)
 # CELIAC   CONTROL 
 #    102     103

US.ID <- rownames(metadata.US)

US.contig <- contig %>%
             filter(V1 %in% US.ID)


############################ US C-section ################################

metadata.US.C <- metadata.US %>%
                 filter(Delivery.Mode == "C-Section")
# 18 24

table(metadata.US.C$Dx.Status)
# CONTROL 
#  18 


US.C.ID <- rownames(metadata.US.C)

US.C.contig <- contig %>%
             filter(V1 %in% US.C.ID)


############################ US C-section breastfeeding ################################

metadata.US.C.breast <- metadata.US.C %>%
                        filter(feeding_first_year == "Breast_fed")

# 0 24

US.C.breast.ID <- rownames(metadata.US.C.breast)


US.C.breast.contig <- contig %>%
             filter(V1 %in% US.C.breast.ID)


############################ US C-section Formula ################################

metadata.US.C.Formula <- metadata.US.C %>%
                        filter(feeding_first_year == "Formula")

# 9 24

US.C.Formula.ID <- rownames(metadata.US.C.Formula)

US.C.Formula.contig <- contig %>%
                       filter(V1 %in% US.C.Formula.ID)

############################ US C-section Formula 12M ################################

metadata.US.C.Formula.12M <- metadata.US.C.Formula %>%
                             filter(month == "12")

# 1 24

US.C.Formula.12M.ID <- rownames(metadata.US.C.Formula.12M)


US.C.Formula.12M.contig <- contig %>%
                           filter(V1 %in% US.C.Formula.12M.ID)

############################ US C-section Formula 18M ################################

metadata.US.C.Formula.18M <- metadata.US.C.Formula %>%
                             filter(month == "18")
# 1 24


US.C.Formula.18M.ID <- rownames(metadata.US.C.Formula.18M)

US.C.Formula.18M.contig <- contig %>%
                           filter(V1 %in% US.C.Formula.18M.ID)


############################ US C-section Formula 24M ################################

metadata.US.C.Formula.24M <- metadata.US.C.Formula %>%
                             filter(month == "24")

# 1 24

US.C.Formula.24M.ID <- rownames(metadata.US.C.Formula.24M)

US.C.Formula.24M.contig <- contig %>%
                           filter(V1 %in% US.C.Formula.24M.ID)

############################ US C-section Formula 30M+ ################################

metadata.US.C.Formula.30M.older <- metadata.US.C.Formula %>%
                                   filter(month %in% c(30,36,48,60,72,84))
# 6 24

US.C.Formula.30M.older.ID <- rownames(metadata.US.C.Formula.30M.older)

US.C.Formula.30M.older.contig <- contig %>%
                                 filter(V1 %in% US.C.Formula.30M.older.ID)


############################ US C-section Mix ################################

metadata.US.C.Mix <- metadata.US.C %>%
                        filter(feeding_first_year == "Breastmilk_and_formula")
# 9 24

US.C.Mix.ID <- rownames(metadata.US.C.Mix)

US.C.Mix.contig <- contig %>%
                   filter(V1 %in% US.C.Mix.ID)

############################ US C-section Mix 12M ################################

metadata.US.C.Mix.12M <- metadata.US.C.Mix %>%
                             filter(month == "12")
# 1 24

US.C.Mix.12M.ID <- rownames(metadata.US.C.Mix.12M)


US.C.Mix.12M.contig <- contig %>%
                       filter(V1 %in% US.C.Mix.12M.ID)

############################ US C-section Mix 18M ################################

metadata.US.C.Mix.18M <- metadata.US.C.Mix %>%
                             filter(month == "18")
# 1 24

US.C.Mix.18M.ID <- rownames(metadata.US.C.Mix.18M)

US.C.Mix.18M.contig <- contig %>%
                      filter(V1 %in% US.C.Mix.18M.ID)


############################ US C-section Mix 24M ################################

metadata.US.C.Mix.24M <- metadata.US.C.Mix %>%
                             filter(month == "24")
# 1 24


US.C.Mix.24M.ID <- rownames(metadata.US.C.Mix.24M)

US.C.Mix.24M.contig <- contig %>%
                       filter(V1 %in% US.C.Mix.24M.ID)



############################ US C-section Mix 30M+ ################################

metadata.US.C.Mix.30M.older <- metadata.US.C.Mix %>%
                                   filter(month %in% c(30,36,48,60,72,84))
# 6 24

US.C.Mix.30M.older.ID <- rownames(metadata.US.C.Mix.30M.older)

US.C.Mix.30M.older.contig <- contig %>%
                             filter(V1 %in% US.C.Mix.30M.older.ID)


############################ US Virginal ################################

metadata.US.V <- metadata.US %>%
                 filter(Delivery.Mode == "Vaginal")
# 187  24

table(metadata.US.V$Dx.Status)

US.V.ID <- rownames(metadata.US.V)

US.V.contig <- contig %>%
               filter(V1 %in% US.V.ID)

############################ US Virginal breastfeeding ################################

metadata.US.V.breast <- metadata.US.V %>%
                        filter(feeding_first_year == "Breast_fed")
# 86 24

table(metadata.US.V.breast$Dx.Status)

US.V.breast.ID <- rownames(metadata.US.V.breast)


US.V.breast.contig <- contig %>%
                      filter(V1 %in% US.V.breast.ID)

############################ US Virginal breastfeeding 12M ################################

metadata.US.V.breast.12M <- metadata.US.V.breast %>%
                             filter(month == "12")
# 14 24

table(metadata.US.V.breast.12M$Dx.Status)


US.V.breast.12M.ID <- rownames(metadata.US.V.breast.12M)

US.V.breast.12M.contig <- contig %>%
                      filter(V1 %in% US.V.breast.12M.ID)



############################ US Virginal breastfeeding 18M ################################

metadata.US.V.breast.18M <- metadata.US.V.breast %>%
                             filter(month == "18")
# 14 24

table(metadata.US.V.breast.18M$Dx.Status)
 # CELIAC CONTROL 
 #  6       8 

US.V.breast.18M.ID <- rownames(metadata.US.V.breast.18M)

US.V.breast.18M.contig <- contig %>%
                      filter(V1 %in% US.V.breast.18M.ID)

############################ US Virginal breastfeeding 24M ################################

metadata.US.V.breast.24M <- metadata.US.V.breast %>%
                             filter(month == "24")
# 13 24

table(metadata.US.V.breast.24M$Dx.Status)

US.V.breast.24M.ID <- rownames(metadata.US.V.breast.24M)

US.V.breast.24M.contig <- contig %>%
                      filter(V1 %in% US.V.breast.24M.ID)

############################ US Virginal breastfeeding 30M+ ################################

metadata.US.V.breast.30M.older <- metadata.US.V.breast %>%
                                   filter(month %in% c(30,36,48,54,60,72,84))
# 45 24


table(metadata.US.V.breast.30M.older$Dx.Status)

US.V.breast.30M.older.ID <- rownames(metadata.US.V.breast.30M.older)

US.V.breast.30M.older.contig <- contig %>%
                      filter(V1 %in% US.V.breast.30M.older.ID)

############################ US Virginal Formula ################################

metadata.US.V.Formula <- metadata.US.V %>%
                        filter(feeding_first_year == "Formula")
# 18 24


table(metadata.US.V.Formula$Dx.Status)

US.V.Formula.ID <- rownames(metadata.US.V.Formula)


US.V.Formula.contig <- contig %>%
                      filter(V1 %in% US.V.Formula.ID)

############################ US Virginal Formula 12M ################################

metadata.US.V.Formula.12M <- metadata.US.V.Formula %>%
                             filter(month == "12")
# 4 24

table(metadata.US.V.Formula.12M$Dx.Status)
 # CELIAC CONTROL 
 #      2       2 

US.V.Formula.12M.ID <- rownames(metadata.US.V.Formula.12M)


US.V.Formula.12M.contig <- contig %>%
                      filter(V1 %in% US.V.Formula.12M.ID)

############################ US Virginal Formula 18M ################################

metadata.US.V.Formula.18M <- metadata.US.V.Formula %>%
                             filter(month == "18")
# 3 24

table(metadata.US.V.Formula.18M$Dx.Status)
 # CELIAC CONTROL 
 #   1       2

US.V.Formula.18M.ID <- rownames(metadata.US.V.Formula.18M)


US.V.Formula.18M.contig <- contig %>%
                      filter(V1 %in% US.V.Formula.18M.ID)

############################ US Virginal Formula 24M ################################

metadata.US.V.Formula.24M <- metadata.US.V.Formula %>%
                             filter(month == "24")
# 4 24

table(metadata.US.V.Formula.24M$Dx.Status)

US.V.Formula.24M.ID <- rownames(metadata.US.V.Formula.24M)


US.V.Formula.24M.contig <- contig %>%
                      filter(V1 %in% US.V.Formula.24M.ID)


############################ US Virginal Formula 30M+ ################################

metadata.US.V.Formula.30M.older <- metadata.US.V.Formula %>%
                                   filter(month %in% c(30,36,48,60,72,84))
# 7 24

table(metadata.US.V.Formula.30M.older$Dx.Status)
 # CELIAC CONTROL 
 #  3       4

US.V.Formula.30M.older.ID <- rownames(metadata.US.V.Formula.30M.older)


US.V.Formula.30M.older.contig <- contig %>%
                      filter(V1 %in% US.V.Formula.30M.older.ID)



############################ US Virginal Mix ################################

metadata.US.V.Mix <- metadata.US.V %>%
                        filter(feeding_first_year == "Breastmilk_and_formula")

# 83 24

table(metadata.US.V.Mix$Dx.Status)
 # CELIAC CONTROL 
 #  55      28 


US.V.Mix.ID <- rownames(metadata.US.V.Mix)


US.V.Mix.contig <- contig %>%
                      filter(V1 %in% US.V.Mix.ID)


############################ US Virginal Mix 12M ################################

metadata.US.V.Mix.12M <- metadata.US.V.Mix %>%
                             filter(month == "12")
# 14 24


table(metadata.US.V.Mix.12M$Dx.Status)
 # CELIAC CONTROL 
 #   9       5

US.V.Mix.12M.ID <-rownames(metadata.US.V.Mix.12M)


US.V.Mix.12M.contig <- contig %>%
                      filter(V1 %in% US.V.Mix.12M.ID)

############################ US Virginal Mix 18M ################################

metadata.US.V.Mix.18M <- metadata.US.V.Mix %>%
                             filter(month == "18")
# 14 24

table(metadata.US.V.Mix.18M$Dx.Status)
 # CELIAC CONTROL 
 #  9       5

US.V.Mix.18M.ID <- rownames(metadata.US.V.Mix.18M)


US.V.Mix.18M.contig <- contig %>%
                      filter(V1 %in% US.V.Mix.18M.ID)

############################ US Virginal Mix 24M ################################

metadata.US.V.Mix.24M <- metadata.US.V.Mix %>%
                             filter(month == "24")
# 13 24


table(metadata.US.V.Mix.24M$Dx.Status)
 # CELIAC CONTROL 
 #  8       5 

US.V.Mix.24M.ID <- rownames(metadata.US.V.Mix.24M)


US.V.Mix.24M.contig <- contig %>%
                      filter(V1 %in% US.V.Mix.24M.ID)

############################ US Virginal Mix 30M+ ################################

metadata.US.V.Mix.30M.older <- metadata.US.V.Mix %>%
                                   filter(month %in% c(30,36,48,54,60,72,84))
# 42 24

table(metadata.US.V.Mix.30M.older$Dx.Status)
 # CELIAC CONTROL 
 # 29      13 

US.V.Mix.30M.older.ID <- rownames(metadata.US.V.Mix.30M.older)


US.V.Mix.30M.older.contig <- contig %>%
                      filter(V1 %in% US.V.Mix.30M.older.ID)


```


ITALY
```{r}

############################ ITALY ################################

metadata.Italy <- metadata %>%
               filter(Country == "ITALY")
# 130  24

table(metadata.Italy$Dx.Status)
 # CELIAC CONTROL 
 #  63      67 

Italy.ID <- rownames(metadata.Italy)

Italy.contig <- contig %>%
                      filter(V1 %in% Italy.ID)

############################ Italy C-section ################################

metadata.Italy.C <- metadata.Italy %>%
                 filter(Delivery.Mode == "C-Section")
# 78 24

table(metadata.Italy.C$Dx.Status)
 # CELIAC CONTROL 
 # 37      41 

Italy.C.ID <- rownames(metadata.Italy.C)

Italy.C.contig <- contig %>%
                      filter(V1 %in% Italy.C.ID)

############################ Italy C-section breastfeeding ################################

metadata.Italy.C.breast <- metadata.Italy.C %>%
                        filter(feeding_first_year == "Breast_fed")

# 17 24

table(metadata.Italy.C.breast$Dx.Status)
 # CELIAC CONTROL 
 #  9       8 


Italy.C.breast.ID <- rownames(metadata.Italy.C.breast)

Italy.C.breast.contig <- contig %>%
                      filter(V1 %in% Italy.C.breast.ID)

############################ Italy C-section breastfeeding 12M ################################

metadata.Italy.C.breast.12M <- metadata.Italy.C.breast %>%
                             filter(month == "12")

# 5 24

table(metadata.Italy.C.breast.12M$Dx.Status)
 # CELIAC CONTROL 
 #      2       3 

Italy.C.breast.12M.ID <- rownames(metadata.Italy.C.breast.12M)

Italy.C.breast.12M.contig <- contig %>%
                      filter(V1 %in% Italy.C.breast.12M.ID)

############################ Italy C-section breastfeeding 18M ################################

metadata.Italy.C.breast.18M <- metadata.Italy.C.breast %>%
                             filter(month == "18")

# 4 24

table(metadata.Italy.C.breast.18M$Dx.Status)
 # CELIAC CONTROL 
 #      1       3 

Italy.C.breast.18M.ID <- rownames(metadata.Italy.C.breast.18M)


Italy.C.breast.18M.contig <- contig %>%
                      filter(V1 %in% Italy.C.breast.18M.ID)

############################ Italy C-section breastfeeding 24M ################################

metadata.Italy.C.breast.24M <- metadata.Italy.C.breast %>%
                             filter(month == "24")

# 2 24

table(metadata.Italy.C.breast.24M$Dx.Status)
 # CELIAC CONTROL 
 #  1       1 

Italy.C.breast.24M.ID <- rownames(metadata.Italy.C.breast.24M)

Italy.C.breast.24M.contig <- contig %>%
                      filter(V1 %in% Italy.C.breast.24M.ID)

############################ Italy C-section breastfeeding 30M+ ################################

metadata.Italy.C.breast.30M.older <- metadata.Italy.C.breast %>%
                                   filter(month %in% c(30,36,48,60,72,84))
# 6 24

table(metadata.Italy.C.breast.30M.older$Dx.Status)

Italy.C.breast.30M.older.ID <- rownames(metadata.Italy.C.breast.30M.older)


Italy.C.breast.30M.older.contig <- contig %>%
                      filter(V1 %in% Italy.C.breast.30M.older.ID)

############################ Italy C-section Formula ################################

metadata.Italy.C.Formula <- metadata.Italy.C %>%
                        filter(feeding_first_year == "Formula")

# 21 24

table(metadata.Italy.C.Formula$Dx.Status)
 # CELIAC CONTROL 
 #     10      11

Italy.C.Formula.ID <- rownames(metadata.Italy.C.Formula)


Italy.C.Formula.contig <- contig %>%
                      filter(V1 %in% Italy.C.Formula.ID)

############################ Italy C-section Formula 12M ################################

metadata.Italy.C.Formula.12M <- metadata.Italy.C.Formula %>%
                             filter(month == "12")

# 7 24

table(metadata.Italy.C.Formula.12M$Dx.Status)
 # CELIAC CONTROL 
 #   4       3

Italy.C.Formula.12M.ID <- rownames(metadata.Italy.C.Formula.12M)


Italy.C.Formula.12M.contig <- contig %>%
                      filter(V1 %in% Italy.C.Formula.12M.ID)

############################ Italy C-section Formula 18M ################################

metadata.Italy.C.Formula.18M <- metadata.Italy.C.Formula %>%
                             filter(month == "18")
# 6 24

table(metadata.Italy.C.Formula.18M$Dx.Status)
 # CELIAC CONTROL 
 # 4       2 

Italy.C.Formula.18M.ID <- rownames(metadata.Italy.C.Formula.18M)


Italy.C.Formula.18M.contig <- contig %>%
                      filter(V1 %in% Italy.C.Formula.18M.ID)


############################ Italy C-section Formula 24M ################################

metadata.Italy.C.Formula.24M <- metadata.Italy.C.Formula %>%
                             filter(month == "24")

# 3 24

table(metadata.Italy.C.Formula.24M$Dx.Status)
 # CELIAC CONTROL 
 #  1       2 

Italy.C.Formula.24M.ID <- rownames(metadata.Italy.C.Formula.24M)


Italy.C.Formula.24M.contig <- contig %>%
                      filter(V1 %in% Italy.C.Formula.24M.ID)

############################ Italy C-section Formula 30M+ ################################

metadata.Italy.C.Formula.30M.older <- metadata.Italy.C.Formula %>%
                                   filter(month %in% c(30,36,48,60,72,84))
# 5 24

table(metadata.Italy.C.Formula.30M.older$Dx.Status)
 # CELIAC CONTROL 
 #  1       4

Italy.C.Formula.30M.older.ID <- rownames(metadata.Italy.C.Formula.30M.older)

Italy.C.Formula.30M.older.contig <- contig %>%
                      filter(V1 %in% Italy.C.Formula.30M.older.ID)

############################ Italy C-section Mix ################################

metadata.Italy.C.Mix <- metadata.Italy.C %>%
                        filter(feeding_first_year == "Breastmilk_and_formula")
# 36 24

table(metadata.Italy.C.Mix$Dx.Status)
 # CELIAC CONTROL 
 #     16      20 

Italy.C.Mix.ID <- rownames(metadata.Italy.C.Mix)

Italy.C.Mix.contig <- contig %>%
                      filter(V1 %in% Italy.C.Mix.ID)

############################ Italy C-section Mix 12M ################################

metadata.Italy.C.Mix.12M <- metadata.Italy.C.Mix %>%
                             filter(month == "12")
# 9 24

table(metadata.Italy.C.Mix.12M$Dx.Status)
 # CELIAC CONTROL 
 #  4       5 

Italy.C.Mix.12M.ID <- rownames(metadata.Italy.C.Mix.12M)

Italy.C.Mix.12M.contig <- contig %>%
                      filter(V1 %in% Italy.C.Mix.12M.ID)

############################ Italy C-section Mix 18M ################################

metadata.Italy.C.Mix.18M <- metadata.Italy.C.Mix %>%
                             filter(month == "18")
# 9 24

table(metadata.Italy.C.Mix.18M$Dx.Status)
 # CELIAC CONTROL 
 #    4       5 

Italy.C.Mix.18M.ID <- rownames(metadata.Italy.C.Mix.18M)


Italy.C.Mix.18M.contig <- contig %>%
                      filter(V1 %in% Italy.C.Mix.18M.ID)

############################ Italy C-section Mix 24M ################################

metadata.Italy.C.Mix.24M <- metadata.Italy.C.Mix %>%
                             filter(month == "24")
# 5 24

table(metadata.Italy.C.Mix.24M$Dx.Status)
 # CELIAC CONTROL 
 # 2       3 

Italy.C.Mix.24M.ID <- rownames(metadata.Italy.C.Mix.24M)

Italy.C.Mix.24M.contig <- contig %>%
                      filter(V1 %in% Italy.C.Mix.24M.ID)

############################ Italy C-section Mix 30M+ ################################

metadata.Italy.C.Mix.30M.older <- metadata.Italy.C.Mix %>%
                                   filter(month %in% c(30,36,48,60,72,84))
# 13 24

table(metadata.Italy.C.Mix.30M.older$Dx.Status)
 # CELIAC CONTROL 
 #    6       7

Italy.C.Mix.30M.older.ID <- rownames(metadata.Italy.C.Mix.30M.older)

Italy.C.Mix.30M.older.contig <- contig %>%
                      filter(V1 %in% Italy.C.Mix.30M.older.ID)

############################ Italy Virginal ################################

metadata.Italy.V <- metadata.Italy %>%
                 filter(Delivery.Mode == "Vaginal")
# 52  24

table(metadata.Italy.V$Dx.Status)
 # CELIAC CONTROL 
 #  26      26 

Italy.V.ID <- rownames(metadata.Italy.V)


Italy.V.contig <- contig %>%
                      filter(V1 %in% Italy.V.ID)

############################ Italy Virginal breastfeeding ################################

metadata.Italy.V.breast <- metadata.Italy.V %>%
                        filter(feeding_first_year == "Breast_fed")
# 18 24

table(metadata.Italy.V.breast$Dx.Status)
 # CELIAC CONTROL 
 #  7      11 

Italy.V.breast.ID <- rownames(metadata.Italy.V.breast)


Italy.V.breast.contig <- contig %>%
                      filter(V1 %in% Italy.V.breast.ID)

############################ Italy Virginal breastfeeding 12M ################################

metadata.Italy.V.breast.12M <- metadata.Italy.V.breast %>%
                             filter(month == "12")
# 5 24

table(metadata.Italy.V.breast.12M$Dx.Status)
 # CELIAC CONTROL 
 #   2       3 

Italy.V.breast.12M.ID <- rownames(metadata.Italy.V.breast.12M)

Italy.V.breast.12M.contig <- contig %>%
                      filter(V1 %in% Italy.V.breast.12M.ID)

############################ Italy Virginal breastfeeding 18M ################################

metadata.Italy.V.breast.18M <- metadata.Italy.V.breast %>%
                             filter(month == "18")
# 5 24

table(metadata.Italy.V.breast.18M$Dx.Status)
 # CELIAC CONTROL 
 # 2       3 

Italy.V.breast.18M.ID <- rownames(metadata.Italy.V.breast.18M)

Italy.V.breast.18M.contig <- contig %>%
                      filter(V1 %in% Italy.V.breast.18M.ID)

############################ Italy Virginal breastfeeding 24M ################################

metadata.Italy.V.breast.24M <- metadata.Italy.V.breast %>%
                             filter(month == "24")
# 3 24

table(metadata.Italy.V.breast.24M$Dx.Status)
 # CELIAC CONTROL 
 #      1       2 

Italy.V.breast.24M.ID <- rownames(metadata.Italy.V.breast.24M)


Italy.V.breast.24M.contig <- contig %>%
                      filter(V1 %in% Italy.V.breast.24M.ID)


############################ Italy Virginal breastfeeding 30M+ ################################

metadata.Italy.V.breast.30M.older <- metadata.Italy.V.breast %>%
                                   filter(month %in% c(30,36,48,54,60,72,84))
# 5 24

table(metadata.Italy.V.breast.30M.older$Dx.Status)
 # CELIAC CONTROL 
 #      2       3 

Italy.V.breast.30M.older.ID <- rownames(metadata.Italy.V.breast.30M.older)

Italy.V.breast.30M.older.contig <- contig %>%
                      filter(V1 %in% Italy.V.breast.30M.older.ID)

############################ Italy Virginal Formula ################################

metadata.Italy.V.Formula <- metadata.Italy.V %>%
                        filter(feeding_first_year == "Formula")
# 11 24

table(metadata.Italy.V.Formula$Dx.Status)
 # CELIAC CONTROL 
 #      2       9 

Italy.V.Formula.ID <- rownames(metadata.Italy.V.Formula)


Italy.V.Formula.contig <- contig %>%
                      filter(V1 %in% Italy.V.Formula.ID)

############################ Italy Virginal Formula 12M ################################

metadata.Italy.V.Formula.12M <- metadata.Italy.V.Formula %>%
                             filter(month == "12")
# 3 24

table(metadata.Italy.V.Formula.12M$Dx.Status)
 # CELIAC CONTROL 
 #  1       2 

Italy.V.Formula.12M.ID <- rownames(metadata.Italy.V.Formula.12M)


Italy.V.Formula.12M.contig <- contig %>%
                      filter(V1 %in% Italy.V.Formula.12M.ID)

############################ Italy Virginal Formula 18M ################################

metadata.Italy.V.Formula.18M <- metadata.Italy.V.Formula %>%
                             filter(month == "18")
# 3 24

table(metadata.Italy.V.Formula.18M$Dx.Status)
 # CELIAC CONTROL 
 #   1       2 

Italy.V.Formula.18M.ID <- rownames(metadata.Italy.V.Formula.18M)


Italy.V.Formula.18M.contig <- contig %>%
                      filter(V1 %in% Italy.V.Formula.18M.ID)

############################ Italy Virginal Formula 24M ################################

metadata.Italy.V.Formula.24M <- metadata.Italy.V.Formula %>%
                             filter(month == "24")
# 2 24

table(metadata.Italy.V.Formula.24M$Dx.Status)
# CONTROL 
#   2 

Italy.V.Formula.24M.ID <- rownames(metadata.Italy.V.Formula.24M)

Italy.V.Formula.24M.contig <- contig %>%
                      filter(V1 %in% Italy.V.Formula.24M.ID)

############################ Italy Virginal Formula 30M+ ################################

metadata.Italy.V.Formula.30M.older <- metadata.Italy.V.Formula %>%
                                   filter(month %in% c(30,36,48,60,72,84))
# 3 24

table(metadata.Italy.V.Formula.30M.older$Dx.Status)
# CONTROL 
#   3 

Italy.V.Formula.30M.older.ID <- rownames(metadata.Italy.V.Formula.30M.older)

Italy.V.Formula.30M.older.contig <- contig %>%
                      filter(V1 %in% Italy.V.Formula.30M.older.ID)


############################ Italy Virginal Mix ################################

metadata.Italy.V.Mix <- metadata.Italy.V %>%
                        filter(feeding_first_year == "Breastmilk_and_formula")

# 10 24

table(metadata.Italy.V.Mix$Dx.Status)
 # CELIAC CONTROL 
 #      4       6 

Italy.V.Mix.ID <- rownames(metadata.Italy.V.Mix)


Italy.V.Mix.contig <- contig %>%
                      filter(V1 %in% Italy.V.Mix.ID)

############################ Italy Virginal Mix 12M ################################

metadata.Italy.V.Mix.12M <- metadata.Italy.V.Mix %>%
                             filter(month == "12")
# 3 24

table(metadata.Italy.V.Mix.12M$Dx.Status)
 # CELIAC CONTROL 
 #  1       2 

Italy.V.Mix.12M.ID <- rownames(metadata.Italy.V.Mix.12M)

Italy.V.Mix.12M.contig <- contig %>%
                      filter(V1 %in% Italy.V.Mix.12M.ID)

############################ Italy Virginal Mix 18M ################################

metadata.Italy.V.Mix.18M <- metadata.Italy.V.Mix %>%
                             filter(month == "18")
# 3 24

table(metadata.Italy.V.Mix.18M$Dx.Status)
 # CELIAC CONTROL 
 #  1       2 

Italy.V.Mix.18M.ID <- rownames(metadata.Italy.V.Mix.18M)


Italy.V.Mix.18M.contig <- contig %>%
                      filter(V1 %in% Italy.V.Mix.18M.ID)

############################ Italy Virginal Mix 24M ################################

metadata.Italy.V.Mix.24M <- metadata.Italy.V.Mix %>%
                             filter(month == "24")
# 2 24

table(metadata.Italy.V.Mix.24M$Dx.Status)
 # CELIAC CONTROL 
 #  1       1 

Italy.V.Mix.24M.ID <- rownames(metadata.Italy.V.Mix.24M)

Italy.V.Mix.24M.contig <- contig %>%
                      filter(V1 %in% Italy.V.Mix.24M.ID)

############################ Italy Virginal Mix 30M+ ################################

metadata.Italy.V.Mix.30M.older <- metadata.Italy.V.Mix %>%
                                   filter(month %in% c(30,36,48,54,60,72,84))
# 2 24

table(metadata.Italy.V.Mix.30M.older$Dx.Status)
 # CELIAC CONTROL 
 #    1       1 

Italy.V.Mix.30M.older.ID <- rownames(metadata.Italy.V.Mix.30M.older)

Italy.V.Mix.30M.older.contig <- contig %>%
                      filter(V1 %in% Italy.V.Mix.30M.older.ID)

```


save all the US contig tables into US contig table list
```{r}


US.contig.list <- list("US.contig" = US.contig,
                    "US.C.contig" = US.C.contig,
                    "US.C.Formula.contig" = US.C.Formula.contig,
                    "US.C.Formula.12M.contig" = US.C.Formula.12M.contig,
                    "US.C.Formula.18M.contig" = US.C.Formula.18M.contig,
                    "US.C.Formula.24M.contig" = US.C.Formula.24M.contig,
                    "US.C.Formula.30M.older.contig"= US.C.Formula.30M.older.contig,
                    "US.C.Mix.contig" = US.C.Mix.contig,
                    "US.C.Mix.12M.contig" = US.C.Mix.12M.contig,
                    "US.C.Mix.18M.contig" = US.C.Mix.18M.contig,
                    "US.C.Mix.24M.contig" = US.C.Mix.24M.contig,
                    "US.C.Mix.30M.older.contig" = US.C.Mix.30M.older.contig,
                    "US.V.contig" = US.V.contig,
                    "US.V.breast.contig" = US.V.breast.contig,
                    "US.V.breast.12M.contig" = US.V.breast.12M.contig,
                    "US.V.breast.18M.contig" = US.V.breast.18M.contig,
                    "US.V.breast.24M.contig" = US.V.breast.24M.contig,
                    "US.V.breast.30M.older.contig" = US.V.breast.30M.older.contig,
                    "US.V.Formula.contig" = US.V.Formula.contig,
                    "US.V.Formula.12M.contig" = US.V.Formula.12M.contig,
                    "US.V.Formula.18M.contig" = US.V.Formula.18M.contig,
                    "US.V.Formula.24M.contig" = US.V.Formula.24M.contig,
                    "US.V.Formula.30M.older.contig" = US.V.Formula.30M.older.contig,
                    "US.V.Mix.contig" = US.V.Mix.contig,
                    "US.V.Mix.12M.contig" = US.V.Mix.12M.contig,
                    "US.V.Mix.18M.contig" = US.V.Mix.18M.contig,
                    "US.V.Mix.24M.contig" = US.V.Mix.24M.contig,
                    "US.V.Mix.30M.older.contig" = US.V.Mix.30M.older.contig)



Italy.contig.list <- list("Italy.contig" = Italy.contig,
                    "Italy.C.contig" = Italy.C.contig,
                    "Italy.C.breast.contig" = Italy.C.breast.contig,
                    "Italy.C.breast.12M.contig" = Italy.C.breast.12M.contig,
                    "Italy.C.breast.18M.contig" = Italy.C.breast.18M.contig,
                    "Italy.C.breast.24M.contig" = Italy.C.breast.24M.contig,
                    "Italy.C.breast.30M.older.contig"= Italy.C.breast.30M.older.contig,
                    "Italy.C.Formula.contig" = Italy.C.Formula.contig,
                    "Italy.C.Formula.12M.contig" =Italy.C.Formula.12M.contig,
                    "Italy.C.Formula.18M.contig" = Italy.C.Formula.18M.contig,
                    "Italy.C.Formula.24M.contig" = Italy.C.Formula.24M.contig,
                    "Italy.C.Formula.30M.older.contig"= Italy.C.Formula.30M.older.contig,
                    "Italy.C.Mix.contig" = Italy.C.Mix.contig,
                    "Italy.C.Mix.12M.contig" = Italy.C.Mix.12M.contig,
                    "Italy.C.Mix.18M.contig" = Italy.C.Mix.18M.contig,
                    "Italy.C.Mix.24M.contig" = Italy.C.Mix.24M.contig,
                    "Italy.C.Mix.30M.older.contig" = Italy.C.Mix.30M.older.contig,
                    "Italy.V.contig" = Italy.V.contig,
                    "Italy.V.breast.contig" = Italy.V.breast.contig,
                    "Italy.V.breast.12M.contig" = Italy.V.breast.12M.contig,
                    "Italy.V.breast.18M.contig" = Italy.V.breast.18M.contig,
                    "Italy.V.breast.24M.contig" = Italy.V.breast.24M.contig,
                    "Italy.V.breast.30M.older.contig" = Italy.V.breast.30M.older.contig,
                    "Italy.V.Formula.contig" = Italy.V.Formula.contig,
                    "Italy.V.Formula.12M.contig" = Italy.V.Formula.12M.contig,
                    "Italy.V.Formula.18M.contig" = Italy.V.Formula.18M.contig,
                    "Italy.V.Formula.24M.contig" = Italy.V.Formula.24M.contig,
                    "Italy.V.Formula.30M.older.contig" = Italy.V.Formula.30M.older.contig,
                    "Italy.V.Mix.contig" = Italy.V.Mix.contig,
                    "Italy.V.Mix.12M.contig" = Italy.V.Mix.12M.contig,
                    "Italy.V.Mix.18M.contig" = Italy.V.Mix.18M.contig,
                    "Italy.V.Mix.24M.contig" = Italy.V.Mix.24M.contig,
                    "Italy.V.Mix.30M.older.contig" = Italy.V.Mix.30M.older.contig)



```


Do 75% subgroup mapping
```{r}


for (i in  1:length(US.contig.list)){
  
    print(i)
  
    filename <- names(US.contig.list[i])
    print(filename)
    print(dim(US.contig.list[[i]]))
    print(class(US.contig.list[[i]]))

    filter_and_save_wide_data(genesCoverm = US.contig.list[[i]], 
                              covered_fraction_threshold = 0.75, 
                              output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/subgroups/",
                              fileName = filename)
  
}


for (i in  1:length(Italy.contig.list)){
  
  
    filename <- names(Italy.contig.list[i])

    filter_and_save_wide_data(genesCoverm = Italy.contig.list[[i]], 
                              covered_fraction_threshold = 0.75, 
                              output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/subgroups/",
                              fileName = filename)
  
}


```


load the 75% filtered contig tables
```{r}

# Set the path to your directory
path <- "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/subgroups/"

# List all CSV files in the directory
files <- list.files(path, pattern = "\\.rds", full.names = TRUE)

# Read each file into a list
data_list <- lapply(files, readRDS)

# Optional: name each element of the list with the filename (without extension)
names(data_list) <- tools::file_path_sans_ext(basename(files))


# Modify the data frames: set ORFID column as rownames and remove it from the data
data_list_modified <- lapply(data_list, function(df) {
  # Check if the 'ORFID' column exists in the data frame
  if ("ORFID" %in% colnames(df)) {
    rownames(df) <- df$ORFID  # Set 'ORFID' as row names
    df$ORFID <- NULL          # Remove the 'ORFID' column
  }
  return(df)  # Return the modified data frame
})

```


Check how many contigs left in each subgroup
```{r}

# Create an empty list to store the dimensions
dim_list <- lapply(data_list_modified, dim)

# Optional: Name the new list elements with the same names as the original list
names(dim_list) <- names(data_list)

# Check the result (dimension list)
str(dim_list)


```


DE analysis on each subgroup with DESeq2
```{r}

de_summary_list <- list()
sig_gene_sets <- list()
all_gene_sets <- list()

for (i in 1 : length(data_list_modified)) {

  
  message(sprintf("Processing index %d...", i))
  
  # Wrap the body of the loop, not the loop control itself
  tryCatch({
    
    metadata <- read.csv("~/Handley Lab Dropbox/16S/Celiac/metadata/metadata_final_update_11262024.csv") %>%
      column_to_rownames("sampleName")
    
    count <- t(data_list_modified[[i]]) %>%
      data.frame(.) %>%
      t(.)
    
    print("test1")
    
    ID <- colnames(count)
    
    print("test2")
    
    metadata <- metadata %>%
      filter(rownames(.) %in% ID)
    
    metadata <- metadata[ID, , drop = FALSE]  # safer ordering
    
    metadata$Dx.Status <- factor(metadata$Dx.Status, levels = c("CONTROL", "CELIAC"))
    
    dds <- DESeqDataSetFromMatrix(countData = count + 1,
                                  colData = metadata,
                                  design = ~ Dx.Status)
    
    dds <- DESeq(dds)
    
    res <- results(dds, name = "Dx.Status_CONTROL_vs_CELIAC",
                   contrast = c("Dx.Status", "CONTROL", "CELIAC"))
    
    fileName <- names(data_list_modified)[i]
    print(fileName)
    
    write.csv(res, paste0("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/subgroups/DESeq/", fileName, ".csv"),
              quote = FALSE, col.names = FALSE)
    
    
    all_gene_sets[[fileName]] <-  data.frame(res)
    
    # Filter significant genes
    sig.res <- res %>%
      data.frame(.) %>%
      filter(padj < 0.05)
    
    
    # Calculate number and ratio
    num_sig_genes <- nrow(sig.res)
    total_genes <- sum(!is.na(res$padj))
    sig_ratio <- num_sig_genes / total_genes
    
    # Collect into list
    de_summary_list[[fileName]] <- data.frame(
      File = fileName,
      Significant_Genes = num_sig_genes,
      Total_Genes = total_genes,
      Ratio = sig_ratio
    )

    
    sig.contigs <- rownames(sig.res)
    
    sig_gene_sets[[fileName]] <- sig.contigs
    
    volcano_data <- as.data.frame(res)
    volcano_data$logpval <- -log10(volcano_data$pvalue)
    
    vol.Disease <- ggplot(volcano_data %>% filter(!is.na(padj)),
                          aes(x = log2FoldChange, y = logpval)) +
      geom_point(aes(color = ifelse(padj < 0.05, "significant", "not significant")), alpha = 0.6) +
      theme_minimal() +
      labs(title = paste0(gsub(".contig_wide_rpkm_genes_75Cov", "", fileName), " - Volcano Plot Control vs Disease"),
           x = "Log2 Fold Change", y = "-Log10 P-value") +
      scale_color_manual(values = c("gray", "red"))
    
    ggsave(vol.Disease,
           file = paste0("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/subgroups/DESeq/", fileName, ".pdf"),
           dpi = 600, width = 8, height = 6)
    
  }, error = function(e) {
    message(sprintf("Skipping index %d due to error: %s", i, e$message))
    # No need for 'next' — the function already continues to next iteration
  })
}

# After the loop ends
de_summary_df <- do.call(rbind, de_summary_list)


# Optional: write to CSV
write.csv(de_summary_df,
          file = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/subgroups/DESeq/de_summary.csv",
          row.names = FALSE)


```


ComplexUpset
```{r}

library(UpSetR)
library(ComplexUpset)


# Assuming sig_gene_sets is a named list of gene vectors per subgroup
# Convert to presence/absence binary matrix
all_genes <- unique(unlist(sig_gene_sets))
gene_matrix <- sapply(sig_gene_sets, function(gene_set) all_genes %in% gene_set)
gene_matrix <- as.data.frame(gene_matrix)
rownames(gene_matrix) <- all_genes

head(gene_matrix)

# Make UpSet plot
p.upset <-upset(gene_matrix, intersect = names(sig_gene_sets))

ggsave(p.upset,file = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/subgroups/DESeq/upset_plot.pdf",dpi = 600,width = 35,height = 20)


```



heatmap
```{r}

library(pheatmap)

# Combine all significant genes across all subgroups
all_sig_genes <- unique(unlist(sig_gene_sets))

# Create empty matrix
logFC_matrix <- matrix(NA, nrow = length(all_sig_genes), ncol = length(all_gene_sets),
                       dimnames = list(all_sig_genes, names(all_gene_sets)))

View(logFC_matrix)

# Fill in log2FoldChange values for significant genes
for (subgroup in names(all_gene_sets)) {
  res <- all_gene_sets[[subgroup]]
  res <- as.data.frame(res)
  genes_in_group <- sig_gene_sets[[subgroup]]
  common_genes <- intersect(rownames(res), genes_in_group)
  logFC_matrix[common_genes, subgroup] <- res[common_genes, "log2FoldChange"]
}


logFC_matrix_filled <- logFC_matrix
logFC_matrix_filled[is.na(logFC_matrix_filled)] <- 0

heatmap.de.res <- pheatmap(logFC_matrix_filled,
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         main = "Log2 Fold Change of Significant Genes Across Subgroups",
         fontsize_row = 8,
         fontsize_col = 10)



ggsave(heatmap.de.res,file = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/subgroups/DESeq/deseq_res_heatmap.pdf",dpi = 600,width = 20,height = 49)


```


# prepare subgroup of contig count tables
```{r}

#----------------------------------------------12M-----------------------------------------------------#

contig.12M <- contig %>%
             filter(X1 %like% "_12M")

filter_and_save_wide_data(contig.12M, 
                          covered_fraction_threshold = 0.75, 
                          output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_12M/")
contig.processed.12M <- readRDS("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_12M/wide_rpkm_genes_75Cov.rds") %>%
                 column_to_rownames("ORFID")
contig.12M.count.table_0.75 <- t(contig.processed.12M) 
# 71 1486


#prepare abundance table
contig.12M.abundance.table_0.75 <- merge(metadata,contig.12M.count.table_0.75,by = 0,all.y = TRUE) %>%
                              filter(feeding_first_year != "Unknown") %>%
                              filter(Age.at.Gluten.Introduction..months. != "Unknown")
# 65 1511


#prepare PA table
contig.12M.PA.table_0.75 <- contig.12M.abundance.table_0.75 %>%
                         mutate(across(26:1511, ~ as.integer(. != 0)))

dim(contig.12M.PA.table_0.75)
# 65 1511


# make 12M PA check table
PA.contig.12M.check_0.75 <- contig.12M.PA.table_0.75 %>% 
                           dplyr::select(Row.names,starts_with("virus_"),starts_with("edge")) %>% 
                           pivot_longer(cols = c(starts_with("virus_"),starts_with("edge")),names_to = "Contig",values_to = "PA") %>% 
                           group_by(Contig) %>% 
                           mutate(total_PA = sum(PA)) %>%
                           distinct(Contig,total_PA) %>%
                           arrange(-total_PA)

write.csv(PA.contig.12M.check_0.75,"~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_12M/PA.contig.12M.check_0.75.csv")


# find viruses that exist in only 3% of the samples
# 65 * 0.03 = 1.95
non_rare_contig.12M_0.75_0.03 <- PA.contig.12M.check_0.75 %>% filter(total_PA >=2) %>% pull(Contig)
# 193 contigs


#prevelence filtering on abundance table
contig.12M.abundance.table_0.75_prevFiltered <- contig.12M.abundance.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.12M_0.75_0.03)))
dim(contig.12M.abundance.table_0.75_prevFiltered)
# 65 218

# prevelence filtering on PA table 
contig.12M.PA.table_0.75_prevFiltered <- contig.12M.PA.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.12M_0.75_0.03)))
dim(contig.12M.PA.table_0.75_prevFiltered)
# 65 218



#----------------------------------------------18M-----------------------------------------------------#



contig.18M <- contig %>%
             filter(X1 %like% "_18M")


filter_and_save_wide_data(contig.18M, 
                          covered_fraction_threshold = 0.75, 
                          output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_18M/")

contig.processed.18M <- readRDS("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_18M/wide_rpkm_genes_75Cov.rds") %>%
                 column_to_rownames("ORFID")
contig.18M.count.table_0.75 <- t(contig.processed.18M) 

#prepare abundance table
contig.18M.abundance.table_0.75 <- merge(metadata,contig.18M.count.table_0.75,by = 0,all.y = TRUE) %>%
                              filter(feeding_first_year != "Unknown") %>%
                              filter(Age.at.Gluten.Introduction..months. != "Unknown")

dim(contig.18M.abundance.table_0.75)
# 62 2227


#prepare PA table
contig.18M.PA.table_0.75 <- contig.18M.abundance.table_0.75 %>%
                         mutate(across(26:2227, ~ as.integer(. != 0)))

dim(contig.18M.PA.table_0.75)
# 62 2227

# make 12M PA check table
PA.contig.18M.check_0.75 <- contig.18M.PA.table_0.75 %>% 
                           dplyr::select(Row.names,starts_with("virus_"),starts_with("edge")) %>% 
                           pivot_longer(cols = c(starts_with("virus_"),starts_with("edge")),names_to = "Contig",values_to = "PA") %>% 
                           group_by(Contig) %>% 
                           mutate(total_PA = sum(PA)) %>%
                           distinct(Contig,total_PA) %>%
                           arrange(-total_PA)

write.csv(PA.contig.18M.check_0.75,"~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_18M/PA.contig.18M.check_0.75.csv")


# find viruses that exist in only 3% of the samples
# 62 * 0.03 = 1.86
non_rare_contig.18M_0.75_0.03 <- PA.contig.18M.check_0.75 %>% filter(total_PA >=2) %>% pull(Contig)
# 348 contigs


#prevelence filtering on abundance table
contig.18M.abundance.table_0.75_prevFiltered <- contig.18M.abundance.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.18M_0.75_0.03)))
dim(contig.18M.abundance.table_0.75_prevFiltered)
# 62 373

# prevelence filtering on PA table 
contig.18M.PA.table_0.75_prevFiltered <- contig.18M.PA.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.18M_0.75_0.03)))
dim(contig.18M.PA.table_0.75_prevFiltered)
# 62 373

#----------------------------------------------24M-----------------------------------------------------#

contig.24M <- contig %>%
             filter(X1 %like% "_24M")


filter_and_save_wide_data(contig.24M, 
                          covered_fraction_threshold = 0.75, 
                          output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_24M/")

contig.processed.24M <- readRDS("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_24M/wide_rpkm_genes_75Cov.rds") %>%
                 column_to_rownames("ORFID")
contig.24M.count.table_0.75 <- t(contig.processed.24M) 


#prepare abundance table
contig.24M.abundance.table_0.75 <- merge(metadata,contig.24M.count.table_0.75,by = 0,all.y = TRUE) %>%
                              filter(feeding_first_year != "Unknown") %>%
                              filter(Age.at.Gluten.Introduction..months. != "Unknown")
# 48 2098

#prepare PA table
contig.24M.PA.table_0.75 <- contig.24M.abundance.table_0.75 %>%
                         mutate(across(26:2098, ~ as.integer(. != 0)))

dim(contig.24M.PA.table_0.75)
# 48 2098


# make 12M PA check table
PA.contig.24M.check_0.75 <- contig.24M.PA.table_0.75 %>% 
                           dplyr::select(Row.names,starts_with("virus_"),starts_with("edge")) %>% 
                           pivot_longer(cols = c(starts_with("virus_"),starts_with("edge")),names_to = "Contig",values_to = "PA") %>% 
                           group_by(Contig) %>% 
                           mutate(total_PA = sum(PA)) %>%
                           distinct(Contig,total_PA) %>%
                           arrange(-total_PA)

write.csv(PA.contig.24M.check_0.75,"~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_24M/PA.contig.24M.check_0.75.csv")


# find viruses that exist in only 3% of the samples
# 48 * 0.03 = 1.44
non_rare_contig.24M_0.75_0.03 <- PA.contig.24M.check_0.75 %>% filter(total_PA >=2) %>% pull(Contig)
# 274 contigs


#prevelence filtering on abundance table
contig.24M.abundance.table_0.75_prevFiltered <- contig.24M.abundance.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.24M_0.75_0.03)))
dim(contig.24M.abundance.table_0.75_prevFiltered)
# 48 299

# prevelence filtering on PA table 
contig.24M.PA.table_0.75_prevFiltered <- contig.24M.PA.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.24M_0.75_0.03)))
dim(contig.24M.PA.table_0.75_prevFiltered)
# 48 299





#----------------------------------------------30M-----------------------------------------------------#


contig.30M <- contig %>%
             filter(X1 %like% "_30M")

filter_and_save_wide_data(contig.30M, 
                          covered_fraction_threshold = 0.75, 
                          output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_30M/")

contig.processed.30M <- readRDS("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_30M/wide_rpkm_genes_75Cov.rds") %>%
                 column_to_rownames("ORFID")
contig.30M.count.table_0.75 <- t(contig.processed.30M)


#prepare abundance table
contig.30M.abundance.table_0.75 <- merge(metadata,contig.30M.count.table_0.75,by = 0,all.y = TRUE) %>%
                              filter(feeding_first_year != "Unknown") %>%
                              filter(Age.at.Gluten.Introduction..months. != "Unknown")
# 45 1920


#prepare PA table
contig.30M.PA.table_0.75 <- contig.30M.abundance.table_0.75 %>%
                         mutate(across(26:1920, ~ as.integer(. != 0)))

dim(contig.30M.PA.table_0.75)
# 45 1920


# make 12M PA check table
PA.contig.30M.check_0.75 <- contig.30M.PA.table_0.75 %>% 
                           dplyr::select(Row.names,starts_with("virus_"),starts_with("edge")) %>% 
                           pivot_longer(cols = c(starts_with("virus_"),starts_with("edge")),names_to = "Contig",values_to = "PA") %>% 
                           group_by(Contig) %>% 
                           mutate(total_PA = sum(PA)) %>%
                           distinct(Contig,total_PA) %>%
                           arrange(-total_PA)

write.csv(PA.contig.30M.check_0.75,"~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_30M/PA.contig.30M.check_0.75.csv")


# find viruses that exist in only 3% of the samples
# 45 * 0.03 = 1.35
non_rare_contig.30M_0.75_0.03 <- PA.contig.30M.check_0.75 %>% filter(total_PA >=2) %>% pull(Contig)
# 248 contigs


#prevelence filtering on abundance table
contig.30M.abundance.table_0.75_prevFiltered <- contig.30M.abundance.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.30M_0.75_0.03)))
dim(contig.30M.abundance.table_0.75_prevFiltered)
# 45 273

# prevelence filtering on PA table 
contig.30M.PA.table_0.75_prevFiltered <- contig.30M.PA.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.30M_0.75_0.03)))
dim(contig.30M.PA.table_0.75_prevFiltered)
# 45 273



#----------------------------------------------36M-----------------------------------------------------#


contig.36M <- contig %>%
             filter(X1 %like% "_36M")


filter_and_save_wide_data(contig.36M, 
                          covered_fraction_threshold = 0.75, 
                          output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_36M/")
contig.processed.36M <- readRDS("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_36M/wide_rpkm_genes_75Cov.rds") %>%
                 column_to_rownames("ORFID")
contig.36M.count.table_0.75 <- t(contig.processed.36M)


#prepare abundance table
contig.36M.abundance.table_0.75 <- merge(metadata,contig.36M.count.table_0.75,by = 0,all.y = TRUE) %>%
                              filter(feeding_first_year != "Unknown") %>%
                              filter(Age.at.Gluten.Introduction..months. != "Unknown")
dim(contig.36M.abundance.table_0.75)
# 35 1727

#prepare PA table
contig.36M.PA.table_0.75 <- contig.36M.abundance.table_0.75 %>%
                         mutate(across(26:1727, ~ as.integer(. != 0)))

dim(contig.36M.PA.table_0.75)
# 35 1727


# make 12M PA check table
PA.contig.36M.check_0.75 <- contig.36M.PA.table_0.75 %>% 
                           dplyr::select(Row.names,starts_with("virus_"),starts_with("edge")) %>% 
                           pivot_longer(cols = c(starts_with("virus_"),starts_with("edge")),names_to = "Contig",values_to = "PA") %>% 
                           group_by(Contig) %>% 
                           mutate(total_PA = sum(PA)) %>%
                           distinct(Contig,total_PA) %>%
                           arrange(-total_PA)

write.csv(PA.contig.36M.check_0.75,"~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_36M/PA.contig.36M.check_0.75.csv")


# find viruses that exist in only 3% of the samples
# 35 * 0.03 = 1.05
non_rare_contig.36M_0.75_0.03 <- PA.contig.36M.check_0.75 %>% filter(total_PA >=2) %>% pull(Contig)
# 271 contigs


#prevelence filtering on abundance table
contig.36M.abundance.table_0.75_prevFiltered <- contig.36M.abundance.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.36M_0.75_0.03)))
dim(contig.36M.abundance.table_0.75_prevFiltered)
# 35 196

# prevelence filtering on PA table 
contig.36M.PA.table_0.75_prevFiltered <- contig.36M.PA.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.36M_0.75_0.03)))
dim(contig.36M.PA.table_0.75_prevFiltered)
# 35 196

#----------------------------------------------48M-----------------------------------------------------#

contig.48M <- contig %>%
             filter(X1 %like% "_48M")

filter_and_save_wide_data(contig.48M, 
                          covered_fraction_threshold = 0.75, 
                          output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_48M/")

contig.processed.48M <- readRDS("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_48M/wide_rpkm_genes_75Cov.rds") %>%
                 column_to_rownames("ORFID")
contig.48M.count.table_0.75 <- t(contig.processed.48M)


#prepare abundance table
contig.48M.abundance.table_0.75 <- merge(metadata,contig.48M.count.table_0.75,by = 0,all.y = TRUE) %>%
                              filter(feeding_first_year != "Unknown") %>%
                              filter(Age.at.Gluten.Introduction..months. != "Unknown")
# 33 1509


#prepare PA table
contig.48M.PA.table_0.75 <- contig.48M.abundance.table_0.75 %>%
                         mutate(across(26:1509, ~ as.integer(. != 0)))

dim(contig.48M.PA.table_0.75)
# 33 1509


# make 12M PA check table
PA.contig.48M.check_0.75 <- contig.48M.PA.table_0.75 %>% 
                           dplyr::select(Row.names,starts_with("virus_"),starts_with("edge")) %>% 
                           pivot_longer(cols = c(starts_with("virus_"),starts_with("edge")),names_to = "Contig",values_to = "PA") %>% 
                           group_by(Contig) %>% 
                           mutate(total_PA = sum(PA)) %>%
                           distinct(Contig,total_PA) %>%
                           arrange(-total_PA)

write.csv(PA.contig.48M.check_0.75,"~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_48M/PA.contig.48M.check_0.75.csv")


# find viruses that exist in only 3% of the samples
# 35 * 0.03 = 1.05
non_rare_contig.48M_0.75_0.03 <- PA.contig.48M.check_0.75 %>% filter(total_PA >=2) %>% pull(Contig)
# 181 contigs


#prevelence filtering on abundance table
contig.48M.abundance.table_0.75_prevFiltered <- contig.48M.abundance.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.48M_0.75_0.03)))
dim(contig.48M.abundance.table_0.75_prevFiltered)
# 33 206

# prevelence filtering on PA table 
contig.48M.PA.table_0.75_prevFiltered <- contig.48M.PA.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.48M_0.75_0.03)))
dim(contig.48M.PA.table_0.75_prevFiltered)
# 33 206



#----------------------------------------------60M-----------------------------------------------------#


contig.60M <- contig %>%
             filter(X1 %like% "_60M")

filter_and_save_wide_data(contig.60M, 
                          covered_fraction_threshold = 0.75, 
                          output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_60M/")
contig.processed.60M <- readRDS("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_60M/wide_rpkm_genes_75Cov.rds") %>%
                 column_to_rownames("ORFID")
contig.60M.count.table_0.75 <- t(contig.processed.60M)

#prepare abundance table
contig.60M.abundance.table_0.75 <- merge(metadata,contig.60M.count.table_0.75,by = 0,all.y = TRUE) %>%
                              filter(feeding_first_year != "Unknown") %>%
                              filter(Age.at.Gluten.Introduction..months. != "Unknown")
# 16 1005


#prepare PA table
contig.60M.PA.table_0.75 <- contig.60M.abundance.table_0.75 %>%
                         mutate(across(26:1005, ~ as.integer(. != 0)))

dim(contig.60M.PA.table_0.75)
# 16 1005


# make 12M PA check table
PA.contig.60M.check_0.75 <- contig.60M.PA.table_0.75 %>% 
                           dplyr::select(Row.names,starts_with("virus_"),starts_with("edge")) %>% 
                           pivot_longer(cols = c(starts_with("virus_"),starts_with("edge")),names_to = "Contig",values_to = "PA") %>% 
                           group_by(Contig) %>% 
                           mutate(total_PA = sum(PA)) %>%
                           distinct(Contig,total_PA) %>%
                           arrange(-total_PA)

write.csv(PA.contig.60M.check_0.75,"~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_60M/PA.contig.60M.check_0.75.csv")


# find viruses that exist in only 3% of the samples
# 35 * 0.03 = 1.05
non_rare_contig.60M_0.75_0.03 <- PA.contig.60M.check_0.75 %>% filter(total_PA >=2) %>% pull(Contig)
# 125 contigs


#prevelence filtering on abundance table
contig.60M.abundance.table_0.75_prevFiltered <- contig.60M.abundance.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.60M_0.75_0.03)))
dim(contig.60M.abundance.table_0.75_prevFiltered)
# 16 150

# prevelence filtering on PA table 
contig.60M.PA.table_0.75_prevFiltered <- contig.60M.PA.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.60M_0.75_0.03)))
dim(contig.60M.PA.table_0.75_prevFiltered)
# 16 150





#----------------------------------------------72M-----------------------------------------------------#

contig.6Y <- contig %>%
             filter(X1 %like% "_72M")

filter_and_save_wide_data(contig.6Y, 
                          covered_fraction_threshold = 0.75, 
                          output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_6Y/")

contig.processed.6Y <- readRDS("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_6Y/wide_rpkm_genes_75Cov.rds") %>%
                 column_to_rownames("ORFID")
contig.6Y.count.table_0.75 <- t(contig.processed.6Y)

#prepare abundance table
contig.6Y.abundance.table_0.75 <- merge(metadata,contig.6Y.count.table_0.75,by = 0,all.y = TRUE) %>%
                              filter(feeding_first_year != "Unknown") %>%
                              filter(Age.at.Gluten.Introduction..months. != "Unknown")
# 4 270

#prepare PA table
contig.6Y.PA.table_0.75 <- contig.6Y.abundance.table_0.75 %>%
                         mutate(across(26:270, ~ as.integer(. != 0)))

dim(contig.6Y.PA.table_0.75)
# 4 270


# # make 12M PA check table
# PA.contig.6Y.check_0.75 <- contig.6Y.PA.table_0.75 %>% 
#                            dplyr::select(Row.names,starts_with("virus_"),starts_with("edge")) %>% 
#                            pivot_longer(cols = c(starts_with("virus_"),starts_with("edge")),names_to = "Contig",values_to = "PA") %>% 
#                            group_by(Contig) %>% 
#                            mutate(total_PA = sum(PA)) %>%
#                            distinct(Contig,total_PA) %>%
#                            arrange(-total_PA)
# 
# write.csv(PA.contig.6Y.check_0.75,"~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_6Y/PA.contig.6Y.check_0.75.csv")
# 
# 
# # find viruses that exist in only 3% of the samples
# # 4 * 0.03 = 0.12
# non_rare_contig.6Y_0.75_0.03 <- PA.contig.6Y.check_0.75 %>% filter(total_PA >=0) %>% pull(Contig)
# 
# 
# 
# #prevelence filtering on abundance table
# contig.6Y.abundance.table_0.75_prevFiltered <- contig.6Y.abundance.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.6Y_0.75_0.03)))
# dim(contig.6Y.abundance.table_0.75_prevFiltered)
# 
# 
# # prevelence filtering on PA table 
# contig.6Y.PA.table_0.75_prevFiltered <- contig.6Y.PA.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.6Y_0.75_0.03)))
# dim(contig.6Y.PA.table_0.75_prevFiltered)




#----------------------------------------------84M-----------------------------------------------------#

contig.7Y <- contig %>%
             filter(X1 %like% "_84M")

filter_and_save_wide_data(contig.7Y, 
                          covered_fraction_threshold = 0.75, 
                          output_path = "~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_7Y/")

contig.processed.7Y <- readRDS("~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_7Y/wide_rpkm_genes_75Cov.rds") %>%
                 column_to_rownames("ORFID")
contig.7Y.count.table_0.75 <- t(contig.processed.7Y)

#prepare abundance table
contig.7Y.abundance.table_0.75 <- merge(metadata,contig.7Y.count.table_0.75,by = 0,all.y = TRUE) %>%
                              filter(feeding_first_year != "Unknown") %>%
                              filter(Age.at.Gluten.Introduction..months. != "Unknown")
# 4 258

#prepare PA table
contig.7Y.PA.table_0.75 <- contig.7Y.abundance.table_0.75 %>%
                         mutate(across(26:258, ~ as.integer(. != 0)))

dim(contig.7Y.PA.table_0.75)
# 4 258

# # make 7Y PA check table
# PA.contig.7Y.check_0.75 <- contig.7Y.PA.table_0.75 %>% 
#                            dplyr::select(Row.names,starts_with("virus_"),starts_with("edge")) %>% 
#                            pivot_longer(cols = c(starts_with("virus_"),starts_with("edge")),names_to = "Contig",values_to = "PA") %>% 
#                            group_by(Contig) %>% 
#                            mutate(total_PA = sum(PA)) %>%
#                            distinct(Contig,total_PA) %>%
#                            arrange(-total_PA)
# 
# write.csv(PA.contig.7Y.check_0.75,"~/Handley Lab Dropbox/16S/Celiac/Phage/phage_detection_pipeline_output/Coverm/Contig_7Y/PA.contig.7Y.check_0.75.csv")
# 
# 
# # find viruses that exist in only 3% of the samples
# # 4 * 0.03 = 0.12
# non_rare_contig.7Y_0.75_0.03 <- PA.contig.7Y.check_0.75 %>% filter(total_PA >=2) %>% pull(Contig)
# 
# 
# #prevelence filtering on abundance table
# contig.7Y.abundance.table_0.75_prevFiltered <- contig.7Y.abundance.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.7Y_0.75_0.03)))
# dim(contig.7Y.abundance.table_0.75_prevFiltered)
# 
# 
# # prevelence filtering on PA table 
# contig.7Y.PA.table_0.75_prevFiltered <- contig.7Y.PA.table_0.75 %>% dplyr::select(c(1:25,all_of(non_rare_contig.7Y_0.75_0.03)))
# dim(contig.7Y.PA.table_0.75_prevFiltered)



```


Limma on US and Italy respectively
```{r}


Italy.contig_wide_rpkm_genes_75Cov
US.contig_wide_rpkm_genes_75Cov



# prepare clean metadata
metadata.clean <- contig.abundance.table_0.75[1:25] %>%
                  column_to_rownames("Row.names")
# 314  24

contig.abundance.clean <- contig.abundance.table_0.75 %>%
                           select(c("Row.names",26:6358)) %>%
                           filter(Row.names %in% rownames(metadata.clean)) %>%
                           column_to_rownames("Row.names") %>% 
                           t(.) %>%
                           data.frame(.)
colnames(contig.abundance.clean) <- gsub("X","",colnames(contig.abundance.clean))
# 6333  314

# prepare correct variable format
metadata.clean$feeding_first_year <- factor(metadata.clean$feeding_first_year,levels = c("Breast_fed","Formula","Breastmilk_and_formula"))
# Breast_fed     Formula     Breastmilk_and_formula 
#   121            59                    134 
metadata.clean$HLA.Category <- factor(metadata.clean$HLA.Category,levels = c("Standard Risk","High Risk","Low/No Risk"))
# Standard Risk     High Risk   Low/No Risk 
#      172            94            48
metadata.clean$Country <- factor(metadata.clean$Country,levels = c("ITALY","USA"))
# ITALY   USA 
#   113   201 
metadata.clean$Sex <- factor(metadata.clean$Sex,levels = c("Female","Male"))
# Female   Male 
#    241    73
metadata.clean$Delivery.Mode <- factor(metadata.clean$Delivery.Mode,levels = c("Vaginal","C-Section"))
#  Vaginal   C-Section 
#     222        92
metadata.clean$Age.at.Gluten.Introduction..months. <- factor(metadata.clean$Age.at.Gluten.Introduction..months.,levels = c("4","5","6","7","8","9","10","11","12","15","36"))
 # 4  5  6  7  8  9 10 11 12 15 36 
 # 5 11 61 94 40 33 33 10  2 16  9
metadata.clean$Dx.Status <- factor(metadata.clean$Dx.Status,levels = c("CONTROL","CELIAC"))
# CONTROL  CELIAC 
#     164   150 


model.design <- model.matrix(~ Dx.Status + month + Country + Sex + Age.at.Gluten.Introduction..months. + HLA.Category + feeding_first_year + Delivery.Mode,metadata.clean)
# to avoid errors that caused by contigs that have all 0 values
contig.abundance.clean <- contig.abundance.clean + 1
model.fit <- voomLmFit(contig.abundance.clean,model.design,block= metadata.clean$patientID,plot=FALSE)
model.fit <- eBayes(model.fit)
model.fit$coefficients
model.res <- topTable(model.fit,coef="Dx.StatusCELIAC",number=Inf)
sig.model.res <- topTable(model.fit,coef="Dx.StatusCELIAC",number=Inf) %>% filter(adj.P.Val < 0.05)
dim(sig.model.res)


```
